---
title: "Acute Lymphobastic Leukemia (ALL) GWAS"
author: "Abbas Rizvi"
date: "11/11/2018"
output: pdf_document  
editor_options: 
  chunk_output_type: console
---

```{r ch5_setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(glue)
```

# Introduction
Acute lymphoblastic leukemia (ALL) is a type liquid cancer that originates in the lymphoid line of hematopoietic stem cells. It the most common type of childhood cancer (25% of all cases) and about 60% of cases for those under 20 years old worldwide [@pui2011acute]. Childhood ALL is highly treatable with a combination of standard chemotherapy protocols with a 5-year survival is upwards of 90% [@pui_2015]. Historically adolescents and young adults (AYA) have had poorer prognosis when it came to treatment from ALL, however, over the past several years major improvements have occurred as treatment protocols have shifted from treating AYA with adult regimens towards pediatric regimens [@Huguet_2009;@siegel_2018]. 

While therapeutic regimens have been quite successful, ALL remains the second most common hematological malignancy for which patients will undergo HLA-matched unrelated donor hematopoietic stem cell transplantation (MUD-HSCT) [@DSouza_2017]. The number of transplants among ALL patients has steadily been increasing yearly [@DSouza_2017]. As recently as 2017, there was approximately 6,000 new cases of ALL and approximately 1500 patients underwent MUD-HSCT [@seer; @DSouza_2017]. ALL patients receive HSCT as a curative therapy after reaching complete remission (CR) and disease status at time of diagnosis play a major role in disease prognosis [@pui2011acute]. Patients receive HSCT as a curative therapy after reaching complete remission (CR) and disease status at time of diagnosis play a major role in disease prognosis. Nearly 60% of adult ($\geq 40$ years old) patients with advanced disease die within the first year after transplant [@DSouza_2017]. MUD-HSCT has slightly improved survival in childhood ALL ($\leq 15$ years old) with 40% of advanced disease patients dying in the first year [@DSouza_2017]. These large proportions of patients dying within the first year present a strong need to discover genetic markers that can inform clinicians on patient longevity and/improvements in donor selection.

Here, we use DISCOVeRY-BMT GWAS to identify and characterize germline genetic markers contribution to survival in ALL patients and their matched unrelated donor after MUD-HSCT. While this study is not specifically asking questions about age, over 60% the ALL patient population in DISCOVeRY-BMT are in childhood and AYA populations, presenting an excellent cohort to uncover markers that may contribute to poor outcomes in young and adult patients after transplant. Additionally, targeted therapeutics can be discovered from SNPs, particularly those that alter sequences of transcription factor binding or result in epigenetic remodeling. In this chapter we go through associated regions for recipient survival outcomes when considering either recipient or their unrelated donor genetics. Furthermore, we performed post-GWAS annotation to develop mechanistic and biologically relevant hypotheses of how these SNPs contribute to survival phenotypes. 

# Methods

Chapter 1 provides details on DISCOVeRY BMT cohorts and survival outcome definitions. Chapter 4 describes the clinical covariates included in all analyses and provides an in depth description of analytic methods. Acute lymphoblastic leukemia patients in DISCOVeRY-BMT are cytogenetically normal. We use Cox regression [@cox1972; @Rizvi_2018] to test the significance of genetic variation in: overall survival (OS), disease related mortality (DRM), transplant related mortality (TRM), relapse (REL), progression-free survival (PFS) and TRM subtypes [graft versus host disease (GVHD), infection (INF), and organ failure (OF)]. 

All analyses were performed on one SNP (donors or recipients dosage across all patients) at a time, adjusted for clinically relevant covariates. Meta-analysis was done using a fixed effects model via METAL software [@metal]. Additional QC was performed on the SNPs (i.e. INFO score > 0.8, MAF > 0.05 in both cohorts, required to be same direction between cohorts and heterogeneous SNPs were filtered out).  We define genome-wide significance as $P_{meta} < 5 \times{10}^{-8}$. Reported significant results were filtered $P_{\text{cohort1}} < 5\times{10}^{-5}$ and $P_{\text{cohort2}} < 0.2$. Meta-hazard ratios were filtered $\frac{1}{15} \leq \text{HR}_{meta} \leq 15$.

Post-GWAS annotations were performed by leveraging publicly available databases including HaploReg v4.1 [@haploreg], RegulomeDB [@Boyle_2012], and SNiPA [@snipa]. These annotation repositories were described in detail in Chapter 4. 

# Results

## DISCOVeRY-BMT ALL Summary

```{r, echo=FALSE}
library(tidyverse)
covariate.file <- read_tsv("~/Google Drive/OSU_PHD/DBMT_100d/DBMT_PhenoData_EA_long_allVar_20190223.txt")

covariate.file <- covariate.file %>%
    filter(sample_type=="recipient",
           ALLdummy==1) %>%
    select(cohort, age, intxsurv_1Y, intxrel_1Y, dnrsex, sex, racegp,
           distatD,ncmvc, graftype, bmi_cat, abomatch, alstatpr,
           disease_death_1Y, TRM_1Y, dead_1Y, lfs_1Y,
           rel_1Y, GVHD_death_1Y, infection_1Y, OF_1Y) 

outcome_ct <- function(...) {
    covariate.file %>%
        group_by(cohort) %>%
        count(...) %>%
        mutate(freq=paste0("(", round(n/sum(n),3), ")"),
               desc=paste0(n, "/", sum(n))) %>%
        unite(freq, c("desc", "freq"), sep=" ") %>%
        select(-n) %>%
        gather(key, value, ...) %>%
        spread(cohort, freq) %>%
        filter(value==1)
}   

covs <- vars(disease_death_1Y, TRM_1Y, dead_1Y,
             lfs_1Y, rel_1Y, GVHD_death_1Y, 
             infection_1Y, OF_1Y)

all_stats <- map_dfr(covs, outcome_ct) %>%
    rename(`Cohort 1`=`1`,
           `Cohort 2`=`2`,
           Outcome=key,
           Status=value) %>%
    select(-Status) %>%
    mutate(Outcome=c("DRM", "TRM", "OS", "PFS", "REL", "GVHD", "INF", "OF")) 

all_stats %>%
    knitr:: kable(caption="\\label{tab:all_event_props} Proportion of Events at 1-year in DISCOVeRY-BMT ALL Cohorts", 
                  booktabs=TRUE, 
                  align="c") %>%
    kableExtra::kable_styling(latex_options="striped", 
                              font_size=9,
                              position='center') %>%
    footnote(general_title="",
             general="Proportion of Events (Death or Relapse) are displayed as the sample size of event ($\\\\text{N}_{\\\\text{event}}$) divided by the total number of patients (N) in the ALL cohort. Overlapping events are not shown here.",
             footnote_as_chunk = TRUE,
             threeparttable = TRUE, 
             escape=FALSE)
```

DISCOVeRY-BMT has 483 patients in cohort 1 and 94 patients in cohort 2 (Table \ref{tab:all_event_props}). The proportion of patients that die within the first year (overall survival) is on average 41.35% between both cohorts (Table \ref{tab:all_event_props}). Slightly more patients die from TRM than from DRM at 21.85% and 19.5% between both cohorts, respectively (Table \ref{tab:all_event_props}). About 27% of patients experience relapse after transplant and between 5-7% of patients are dying from TRM subtypes (Table \ref{tab:all_event_props}).

```{r, echo=FALSE}
age_prop_table <- covariate.file %>%
    mutate(age_cat=case_when(age<=15 ~ "Childhood",
                             age>=40 & age < 60~ "Adult",
                             age>=15 & age < 40 ~ "AYA",
                             age>=60 ~ "Senior Adult")) %>% 
    group_by(cohort) %>%
    count(age_cat) %>% 
    mutate(prop=n/sum(n)) %>%
    gather(key, value, age_cat) %>%
    select(-n, -key) %>%
    spread(cohort, prop) %>% 
    dplyr::rename(`Age Category`=value,
           `Cohort 1`=`1`,
           `Cohort 2`=`2`) %>% 
    mutate_at(vars(`Cohort 1`, `Cohort 2`), .funs=funs(round(., 3))) 

age_prop_table <- age_prop_table[c(3,2,1,4),]

age_prop_table %>%
    knitr:: kable(caption="\\label{tab:all_ages} Proportion of ALL Patient Age Groups Stratified by DISCOVeRY-BMT Cohorts",
                  booktabs=TRUE,
                  align="c") %>%
    kableExtra::kable_styling(latex_options="striped", 
                              font_size=9,
                              position='center') %>%
    footnote(general_title="",
             general="Senior Adult: $\\\\geq 60$ years old; 
             Adult: $\\\\geq$ 40 years old and < 60; 
             Adolescent or Young Adult (AYA): $\\\\geq$ 15 and < 40; 
             Childhood < 15",
             footnote_as_chunk = TRUE,
             threeparttable = TRUE,
             escape=FALSE)
```

Age is adjusted as a continuous variable in our Cox regression model, however, for characterization purposes we looked at the age distribution by standard clinical stratification groups. Those groups include senior adults ($\geq 60$), adult ($\geq 40$ years old $< 60$), adolescent and young adults (AYA; $\geq 15$ and $< 40$ years old), and children ($< 15$ years old) (Table \ref{tab:all_ages}). Childhood and AYA comprise 69% of cohort 1 and 63% of cohort 2 (Table \ref{tab:all_ages}). 

## Post-HSCT associations with recipient SNPs

We tested recipient (patient) SNPs for association with OS, PFS, DRM, TRM and GVHD in ALL patients. While all the subgroups of TRM are of interest, for the purpose of this chapter the main focus are on these more frequent outcomes while the rest are exploratory. We detected genome-wide associated regions with DRM and OS outcomes. Some of these regions were also genome-wide significant for patients diagnosed with B-ALL subtype. Identifying such recipient SNPs can be helpful in distinguishing individuals that have a higher risk of mortality following transplant and treatment strategies can be modified accordingly. 

## Recipient SNPs Overall Survival Associations

\begin{figure}
    \centering
    \includegraphics[width=\linewidth, height=3.5in]{~/Desktop/figures/chapter5/R_ALLonly_OS.jpeg}
    \caption[Recipient OS 1 Year Manhattan Plot]{Recipient OS 1 Year Manhattan Plot. The x-axis is chromosome 1-22 and the y-axis is the $-\log_{10}(P_{meta})$. Each dot is a SNP. The red line is genome wide significance at $P_{meta} < 5\times{10}^{-8}$. The blue line is suggestive significance at $P_{meta} < 5\times{10}^{-5}$. Labeled SNPs are associated hits that have passed genome-wide threshold.}
    \label{fig:r_os_1y}
\end{figure}

```{r, echo=FALSE}

joint_res <- readRDS("~/Google Drive/OSU_PHD/DBMT_ALLonly/hits/20190306_suggestive_sig_snps_all_outcomes.rds")

joint_res %>% filter(pt_subset == "R_ALLonly_OS_1y",
                     PVALUE_M < 5e-08, 
                     PVALUE_c1 < 5e-05,
                     PVALUE_c2 < 0.2, 
                     (HR_ADJ < 15 & HR_ADJ > (1/15)), 
                     RSID != "rs117590352") %>%
    arrange(CHR, PVALUE_M) %>%
    unite(`CHR:POS`, c("CHR", "POS"), sep=":") %>% 
    mutate_at(.vars=vars(HR_ADJ, HR_lower_ADJ, HR_upper_ADJ), .funs=funs(round(., 2))) %>%
    mutate_at(.vars=vars(RefPanelAF, SAMP_MAF_c1, SAMP_MAF_c2), .funs=funs(round(., 4))) %>%
    mutate(INFO=round(INFO, 2),
           HR=glue_data(., "{HR_ADJ} ({HR_lower_ADJ}-{HR_upper_ADJ})")) %>%
    select(-PVALUE_c1:-PVALUE_c2, -Direction, -TYPED, -HR_ADJ:-regdb_score, INFO) %>%
    unite(`Ref. Panel/Cohort 1/ Cohort 2 MAF`, c("RefPanelAF", "SAMP_MAF_c1", "SAMP_MAF_c2"), sep="/") %>%
    rename(`HR (95% CI)`=HR,
           `REF/ EFFECT`=REF_ALT,
           `P-value`=PVALUE_M) %>% 
    kable(caption="\\label{tab:r_os_hits} DISCOVeRY-BMT Recipients Overall Survival Associations at 1-Year", 
              digits=60, 
              align='c') %>%
    kableExtra::kable_styling(latex_options="striped",
                              font_size=9, 
                              position='center') %>%
    kableExtra::column_spec(1, width="5em") %>%
    kableExtra::column_spec(2, width="5.5em") %>%
    kableExtra::column_spec(3, width="2em") %>%
    kableExtra::column_spec(4, width="10.5em") %>%
    kableExtra::column_spec(5, width="4.8em") %>%
    kableExtra::column_spec(6, width="8em") %>%
    footnote(general_title="",
             general="All SNPs shown here are imputed. Genomic positions are GRCh37 reference genome.",
             footnote_as_chunk = TRUE,
             threeparttable = TRUE)

```

<!-- OS chr 9 snipa -->

\begin{figure}
    \centering
    \includegraphics[width=\linewidth, height=3.5in]{~/Desktop/figures/chapter5/rs34486026_9_R_ALLonly_OS.png}
    \caption[Regional association plot of recipient SNP rs34486026 with DRM in ALL patients.]{Regional association plot of recipient SNP rs34486026 with DRM in ALL patients. The x-axis is genomic position on chromosome 2 and the y-axis is the $-\log_{10}(P_{meta})$. Sentinel SNP is rs34486026 and colored blue. The region comprises +/- 150 kb window from the sentinel SNP. The red-yellow color gradient represents LD strength. Shapes represent various regulatory or functional effects that the SNP has been reported to have in this region from independent studies.}
    \label{fig:r_os_1y_chr9reg}  
\end{figure}

<!-- OS chr 18 snipa -->

\begin{figure}
    \centering
    \includegraphics[width=\linewidth, height=3.5in]{~/Desktop/figures/chapter5/rs149914041_18_R_ALLonly_OS.png}
    \caption[Regional association plot of recipient SNP rs149914041 with DRM in ALL patients.]{Regional association plot of recipient SNP rs149914041 with DRM in ALL patients. The x-axis is genomic position on chromosome 2 and the y-axis is the $-\log_{10}(P_{meta})$. Sentinel SNP is rs149914041 and colored blue. The region comprises +/- 150 kb window from the sentinel SNP. The red-yellow color gradient represents LD strength. Shapes represent various regulatory or functional effects that the SNP has been reported to have in this region from independent studies.}
    \label{fig:r_os_1y_chr}  
\end{figure}


Association tests of recipient SNPs with overall survival after MUD-HSCT revealed two loci on chromosomes 9 and 18 that reached genome-wide significance (Figure \ref{fig:r_os_1y}; Table \ref{tab:r_os_hits}). 

Association on chromosome 9 was detected for rs34486026 (Figure \ref{fig:r_os_1y_chr9reg}), which changes binding motifs for 6 proteins [@motifs_2014]. rs34486026 is also in LD ($r^2=0.82$) with rs16929156, a protein quantitative trait loci (pQTL) for lysozyme (Lyz) protein. Lyz is a protein with antibacterial activity found in multiple tissues and body fluids. Associated loci on chromosome 18 was rs149914041, which alters motif for Foxa proteins, a subclass of Fox family of transcription factors involved in cell growth, proliferation and longevity. 

We also screened recipient SNPs with suggestive OS associations ($P < 5 \times 10^{-5}$) and detected two regions of interest on chromosomes 6 and 12.

<!-- OS chr 6 snipa -->
\begin{figure}
    \centering
    \includegraphics[width=\linewidth, height=3.5in]{~/Desktop/figures/chapter5/rs9381325_6_R_ALLonly_OS.png}
    \caption[Regional association plot of recipient SNP rs9381325 with DRM in ALL patients.]{Regional association plot of recipient SNP rs9381325 with DRM in ALL patients. The x-axis is genomic position on chromosome 2 and the y-axis is the $-\log_{10}(P_{meta})$. Sentinel SNP is rs9381325 and colored blue. The region comprises +/- 150 kb window from the sentinel SNP. The red-yellow color gradient represents LD strength. Shapes represent various regulatory or functional effects that the SNP has been reported to have in this region from independent studies.}
    \label{fig:r_os_1y_chr6}  
\end{figure}

Identified loci on chromosome 6 were eQTLs for *CDCL5* gene (Figure \ref{fig:r_os_1y_chr6}). *CDCL5* encodes for Cell division cycle 5-like protein, which acts as a transcription activator. Interestingly PRP19-CDC5L complex may also play a role in the response to DNA damage. [@Lei_2000]


<!-- <!-- OS chr 12 snipa -->
<!-- \begin{figure} -->
<!--     \centering -->
<!--     \includegraphics[width=\linewidth, height=3.5in]{~/Desktop/figures/chapter5/rs10843338_12_R_ALLonly_OS.png} -->
<!--     \caption[Regional association plot of recipient SNP rs10843338 with DRM in ALL patients.]{Regional association plot of recipient SNP rs10843338 with DRM in ALL patients. The x-axis is genomic position on chromosome 2 and the y-axis is the $-\log_{10}(P_{meta})$. Sentinel SNP is rs10843338 and colored blue. The region comprises +/- 150 kb window from the sentinel SNP. The red-yellow color gradient represents LD strength. Shapes represent various regulatory or functional effects that the SNP has been reported to have in this region from independent studies.} -->
<!--     \label{fig:r_os_1y_chr12}   -->
<!-- \end{figure} -->

<!-- Region on chromosome 12  -->


## Recipient SNPs Disease Related Mortality Associations 

```{r, echo=FALSE}
library(glue)
joint_res <- readRDS("~/Google Drive/OSU_PHD/DBMT_ALLonly/hits/20190306_suggestive_sig_snps_all_outcomes.rds")

joint_res %>% filter(pt_subset == "R_ALLonly_DRM_1y",
                     PVALUE_M < 5e-08, 
                     PVALUE_c1 < 5e-05,
                     PVALUE_c2 < 0.2, 
                     (HR_ADJ < 15 & HR_ADJ > (1/15)), 
                     RSID != "rs370808742") %>%
    arrange(CHR, PVALUE_M) %>%
    unite(`CHR:POS`, c("CHR", "POS"), sep=":") %>% 
    mutate_at(.vars=vars(HR_ADJ, HR_lower_ADJ, HR_upper_ADJ), .funs=funs(round(., 2))) %>%
    mutate_at(.vars=vars(RefPanelAF, SAMP_MAF_c1, SAMP_MAF_c2), .funs=funs(round(., 4))) %>%
    mutate(INFO=round(INFO, 2),
           HR=glue_data(., "{HR_ADJ} ({HR_lower_ADJ}-{HR_upper_ADJ})")) %>%
    select(-PVALUE_c1:-PVALUE_c2, -Direction, -TYPED, -HR_ADJ:-regdb_score, INFO) %>%
    unite(`Ref. Panel/Cohort 1/ Cohort 2 MAF`, c("RefPanelAF", "SAMP_MAF_c1", "SAMP_MAF_c2"), sep="/") %>%
    rename(`HR (95% CI)`=HR,
           `REF/ EFFECT`=REF_ALT,
           `P-value`=PVALUE_M) %>% 
        kable(caption="\\label{tab:r_drm_hits} DISCOVeRY-BMT recipient genotypes associated with disease related mortality at 1-year.", 
              digits=60, 
              align='c') %>%
    kableExtra::kable_styling(latex_options="striped",
                              font_size=9, 
                              position='center') %>%
    kableExtra::column_spec(1, width="5em") %>%
    kableExtra::column_spec(2, width="5.5em") %>%
    kableExtra::column_spec(3, width="2em") %>%
    kableExtra::column_spec(4, width="10.5em") %>%
    kableExtra::column_spec(5, width="4.8em") %>%
    kableExtra::column_spec(6, width="8em") %>%
    footnote(general_title="",
             general="All SNPs shown are imputed. Genomic positions are GRCh37 reference genome.",
             footnote_as_chunk = TRUE,
             threeparttable = TRUE)
```

\begin{figure}
    \centering
    \includegraphics[width=\linewidth, height=3.5in]{~/Desktop/figures/chapter5/R_ALLonly_DRM.jpeg}
    \caption[Manhattan Plot of Recipient ALL DRM GWAS.]{Manhattan Plot of Recipient ALL DRM GWAS. The x-axis is chromosome 1-22 and the y-axis is the $-\log_{10}(P_{meta})$. Each dot is a SNP. The red line is genome wide significance at $P_{meta} < 5\times{10}^{-8}$. The blue line is suggestive significance at $P_{meta} <  5\times{10}^{-5}$. Labeled SNPs are associated hits that have passed genome-wide threshold.}
    \label{fig:r_drm_1y}  
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=\linewidth, height=3.5in]{~/Desktop/figures/chapter5/rs4663706_2_R_ALLonly_DRM.png}
    \caption[Regional association plot of recipient SNP rs4663706 with DRM in ALL patients.]{Regional association plot of recipient SNP rs4663706 with DRM in ALL patients. The x-axis is genomic position on chromosome 2 and the y-axis is the $-\log_{10}(P_{meta})$. Sentinel SNP is rs4663706 and colored blue. The region comprises +/- 150 kb window from the sentinel SNP. The red-yellow color gradient represents LD strength. Shapes represent various regulatory or functional effects that the SNP has been reported to have in this region from independent studies.}
    \label{fig:r_drm_1y_chr2reg1}  
\end{figure}

We found genome-wide recipient SNP associations ($P_{meta} < 5\times{10}^{-8}$) with DRM on chromosomes 2, 11, and 17 (Figure \ref{fig:r_drm_1y}). 

Among the genome-wide associated (GWA) SNPs on chromosome 2, rs4663706 had the strongest association (Table \ref{tab:r_drm_hits}). Three SNPs (rs55671334, rs4663709, rs76276644) in the region are in LD with rs4663706 at $r^2>0.8$. These SNPs are located in the upstream of *COL6A3* gene and this intergenic region has been predicted as an enhancer region in multiple tissues, including primary hematopoietic stem cells by Roadmap Epigenome project (Figure \ref{fig:r_drm_1y_chr2reg1}). It is important to note however, that all four SNPs including rs4663706 are imputed ($\text{INFO} > 0.8$). Furthermore, while rs55671334, rs4663709, rs76276644 had genome-wide significant $P_{meta}$ values, (Figure \ref{fig:r_drm_1y}) the associations appear to be driven by cohort 1, as $P_{\text{cohort2}} > 0.1$ (Table \ref{tab:r_drm_hits}). Thus, p-values in cohort 2 weaken confidence in the genome-wide association in this region. 

\begin{figure}
    \centering
    \includegraphics[width=\linewidth, height=3.5in]{~/Desktop/figures/chapter5/rs118004702_11_R_ALLonly_DRM.png}
    \caption[Regional association plot of recipient SNP rs118004702 with DRM in ALL patients.]{Regional association plot of recipient SNP rs118004702 with DRM in ALL patients. The x-axis is genomic position on chromosome 2 and the y-axis is the $-\log_{10}(P_{meta})$. Sentinel SNP is rs118004702 and colored blue. The region comprises +/- 150 kb window from the sentinel SNP. The red-yellow color gradient represents LD strength. Shapes represent various regulatory or functional effects that the SNP has been reported to have in this region from independent studies.}
    \label{fig:r_drm_1y_chr11reg}  
\end{figure}

The region on chromosome 11, includes two genome-wide associated SNPs, rs118004702 and rs117517252 ($r^2=0.94$) (Table \ref{tab:r_drm_hits}; Figure \ref{fig:r_drm_1y}; Figure \ref{fig:r_drm_1y_chr11reg}). ENCODE ChIP-Seq experiments report that SNPs in this region alter regulatory motifs on *HNF4* and *TAL1* [@motifs_2014]. *TAL1* has been reported as an oncogenic transcription factor in T-cell ALL subtype [@sanda_2017]. 

\begin{figure}
    \centering
    \includegraphics[width=\linewidth, height=3.5in]{~/Desktop/figures/chapter5/rs149448001_17_R_ALLonly_DRM.png}
    \caption[Regional association plot of recipient SNP rs149448001 with DRM in ALL patients.]{Regional association plot of recipient SNP rs149448001 with DRM in ALL patients. The x-axis is genomic position on chromosome 2 and the y-axis is the $-\log_{10}(P_{meta})$. Sentinel SNP is rs149448001 and colored blue. The region comprises +/- 150 kb window from the sentinel SNP. The red-yellow color gradient represents LD strength. Shapes represent various regulatory or functional effects that the SNP has been reported to have in this region from independent studies.}
    \label{fig:r_drm_1y_chr17reg}  
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=\linewidth, height=3.5in]{~/Desktop/figures/chapter5/rs149448001_motif.png}
    \caption[Recipient DRM Chromosome 17 rs149448001 motif.]{Recipient DRM Chromosome 17 rs149448001 motif. A schematic of chromosome 17 is at the top. The red vertical line it the genomic location that the of this motif. The two panels show BRCA1 binding motif on chromosome 7 from two different human motif databases. rs149448001 is located at position 8 from T to C. The coordinate is highly predicted to be a T, and therefore a C allele will likely alter the binding properties of proteins to this region.}
    \label{fig:r_drm_motif}  
\end{figure}


Chromosome 17 has four recipient SNPs (rs149448001, rs3865256, rs147024703, rs77119354) associated with DRM within the first year transplant (Table \ref{tab:r_drm_hits}; Figure \ref{tab:r_drm_hits}; Figure \ref{fig:r_drm_1y_chr17reg}). All four SNPs are located in the intronic region of CDRT1 and alter binding motifs for various proteins known to play important roles in leukemias or other cancers (i.e.  *BRCA1*, *CDX*, *E2A*, *STAT* and *GATA* families) [@PerezAndreu_2015]. The SNP rs149448001 had the strongest association in this region and significantly changes the binding motif of *BRCA1*, a well known tumor suppressor. In fact rs149448001 changes the only unambiguous 8th position in the motif from T to C (Figure \ref{fig:r_drm_motif}; @motifbreakr).The Roadmap Epigenome Project, places one of the associated SNPs, rs77119354, in an enhancer region in almost all primary T-cell and hematopoietic stem cell lines [@haploreg]. 

\begin{figure}
    \centering
    \includegraphics[width=\linewidth, height=3.5in]{~/Desktop/figures/chapter5/rs10516806_4_R_ALLonly_DRM.png}
    \caption[Regional association plot of recipient SNP rs10516806 with DRM in ALL patients.]{Regional association plot of recipient SNP rs10516806 with DRM in ALL patients. The x-axis is genomic position on chromosome 2 and the y-axis is the $-\log_{10}$(P-value). Sentinel SNP is rs10516806 and colored blue. The region comprises +/- 150 kb window from the sentinel SNP. The red-yellow color gradient represents LD strength. Shapes represent various regulatory or functional effects that the SNP has been reported to have in this region from independent studies.}
    \label{fig:r_drm_1y_chr4reg}  
\end{figure}

We also inspected recipient SNPs with suggestive DRM associations ($P < 5\times{10}^{-5}$) for previously reported regulatory function. This revealed a region on chromosome 4 where blood metabolite levels associations [rs10022462; @shin_2014] (Figure \ref{fig:r_drm_1y_chr4reg}) had been reported. The blood metabolite GWAS hit is in LD with rs10516806 ($r^{2} = 0.82$; $P=1.6\times{10}^{-7}$; HR: 0.474 [0.359-0.627] for T allele) in recipient DRM. Rs10516806 is in perfect LD ($r^2=1$) with rs2869930 which has been identified as protein bound in ChIP-Seq experiments for *FOXA1* (in liver cell line Hep G2) and \textit{ER$\alpha$} (also known as *NR3A1*; in T-47D human breast cancer cells) [@encode_2012]. *FOXA1* is a transcription factor that belongs to the *FOX* family -- of which another member of this family, *FOXP1*, when downregulated is associated with deficient B-cells [@hu_2006; @Wlodarska_2005]. *FOXA1* and *FOXP1* are transcription factors that are known to modulate activity of \textit{ER$\alpha$} in breast cancer [@Ijichi_2013]. It may be plausible to further interrogate *FOXA1* status in patients who die from their original disease. 

rs10516806 and SNPs that are in LD ($r^2$ > 0.82) have enhancer and promoter histone marks in blood and transplant related tissues (i.e. skin, GI). Rs10516806 is associated with chromatin marks (H3K4me1 for primary T cells from peripheral blood). SNPs in this region alter regulatory motifs on *CEBPB* family genes, *ZPF410*, *FOXP3*, *HOXD8*, and *PAX5*. *ZPF410* is a *PAX5* activated gene [@yang_2015]. *PAX5* encodes for transcription factors that are essential for progression of adult B lymphoiesis from early B cell progenitor cells [@nutt_1998; @shah_2013]. This region has been reported to be associated to gene expression or quantitative traits in several studies. The locus rs10516806 is an eQTL and effects *PPM1K* gene expression in liver and breast cell lines [@li_2014]. Furthermore, a recent study *PPM1K* is involved in amino acid catabolism in HSCs and plays a major role in maintaining stem cell status and HSC repopulation in mice bone marrow [@Liu_2018]. This study suggests that *PPM1K* deletion in murine models improve survival [@Liu_2018]. 

While this could manifest different phenotypes in humans, it suggests that investigating the direction (risk or protective) rs10516806 alters *PPM1K* gene expression is reasonable.   

```{r chr4rs10516806_table, echo=FALSE, eval=FALSE}
library(tidytext)
library(tidyverse)
library(glue)

haplo_sig <- readRDS("~/Google Drive/OSU_PHD/DBMT_ALLonly/hits/haplo_results.rds")
joint_res <- readRDS("~/Google Drive/OSU_PHD/DBMT_ALLonly/hits/joint_res.rds")

joint_res_split <- joint_res %>% 
    filter(PVALUE_M < 5e-06, PVALUE_c1 < 0.05 , PVALUE_c2 < 0.05) %>%
    mutate(pt_subset=str_replace(pt_subset, "ALLsubtype_Bcell", "BALL")) %>%  
    separate(pt_subset, c("donor", "disease", "outcome", "interval"), sep="_") %>%
    filter(RSID=="rs10516806")

e_codes <- read_tsv("~/Google Drive/OSU_PHD/DBMT_ALLonly/hits/ecode_epigenome.txt")
e_codes <- e_codes %>%
    mutate(`Standardised epigenome name`=str_replace_all(`Standardised epigenome name`, "\xca", " ")) %>%
    mutate_all(tolower) %>%
    mutate_all(trimws)
e_codes_vec <- e_codes$`Standardised epigenome name`
names(e_codes_vec) <- e_codes$EID

explore_haplo <- function(donor, disease, outcome, interval, haplo_col){
    df <- joint_res_split %>%
        filter(donor==!!donor,
               disease==!!disease,
               outcome==!!outcome,
               interval==!!interval) 
    haplo_df <- haplo_sig %>% 
        filter(query_snp_rsid %in% df$RSID)
    
    switch(haplo_col,
       motifs={
           haplo_df %>%
               filter(Motifs != ".") %>%
               unnest_tokens(motifs, Motifs, token="regex", pattern = ";") %>% 
               na.omit() %>% 
               mutate_all(trimws) %>%
               count(motifs) %>%
               arrange(desc(n))
        },
        proteins={
            haplo_df %>%
                filter(Proteins != ".") %>%
                select(query_snp_rsid, rsid, Proteins) %>%
                unnest_tokens(proteins, Proteins, token="regex", pattern = ";") %>%
                na.omit() %>% 
                separate(proteins, c('cell lines', 'proteins', 'group', 'treatment'), sep=",") %>%
                unite(cell_line_protein, c('cell lines', "proteins"), sep=";") %>%
                mutate_all(trimws) %>%
                count(cell_line_protein) %>%
                arrange(desc(n)) %>%
                separate(cell_line_protein, c("cell_line", "protein"), sep=";")
        },
        chromatin_marks={
            haplo_df %>%
                select(query_snp_rsid, rsid, Chromatin_Marks) %>%
                unnest_tokens(chromatin_marks, Chromatin_Marks, token="regex", pattern = ";") %>%
                na.omit() %>%
                mutate_all(trimws) %>%
                count(chromatin_marks) %>%
                filter(!(chromatin_marks == "none"),
                       (chromatin_marks!= ".")) %>%
                arrange(desc(n)) %>%
                separate(chromatin_marks, c("eid", "chromatin_mark"), sep=",") %>%
                mutate(eid=e_codes_vec[eid])
        },
        promoter={
            haplo_df %>%
                select(query_snp_rsid, 
                       rsid,
                       Promoter_histone_marks) %>%
                unnest_tokens(promoter_histone_marks, 
                              Promoter_histone_marks, token="regex", 
                              pattern = ";") %>%
                na.omit() %>%
                unnest_tokens(Promoter_histone_marks,
                              promoter_histone_marks, token="regex",
                              pattern = ",|\\n") %>%
                mutate_all(trimws) %>%
                count(Promoter_histone_marks) %>%
                filter(!(Promoter_histone_marks == "none"),
                       (Promoter_histone_marks != ".")) %>%
                arrange(desc(n))
        },
        enhancer={
            haplo_df %>%
                select(query_snp_rsid, 
                       rsid,
                       Enhancer_histone_marks) %>%
                unnest_tokens(enhancer_histone_marks,
                              Enhancer_histone_marks, 
                              token="regex", 
                              pattern = ";") %>%
                na.omit() %>%
                unnest_tokens(Enhancer_histone_marks,
                              enhancer_histone_marks,
                              token="regex",
                              pattern = ",|\\n") %>%
                mutate_all(trimws) %>%
                count(Enhancer_histone_marks) %>%
                filter(!(Enhancer_histone_marks == "none"),
                       (Enhancer_histone_marks != ".")) %>%
                arrange(desc(n)) 
        },
       eqtl={
           haplo_df %>%
               select(query_snp_rsid, rsid, eQTL) %>%
               unnest_tokens(eqtl, eQTL, token="regex", pattern = ";") %>%
               filter(eqtl != ".") %>%
               na.omit() %>%
               separate(eqtl, c("gtex", "cell_lines", "gene", "pvalue"), sep=",") %>%
               select(-rsid, -pvalue, -gtex) %>%
               distinct() %>%
               group_by(cell_lines, gene) %>%
               summarize(n=n()) %>%
               ungroup() %>%
               arrange(desc(n))
       },
       gwas={
               haplo_df %>%
                   filter(gwas != ".") %>%
                   select(query_snp_rsid, r2, rsid, gwas)     
           },
       grasp={
           haplo_df %>%
               filter(grasp != ".") %>%
               select(query_snp_rsid, r2, rsid, grasp) %>% 
               mutate(grasp=ifelse(str_detect(grasp, ";"), grasp, paste0(grasp, ";"))) %>%
               unnest_tokens(Grasp, grasp, token="regex", pattern=";") %>%
               separate(Grasp, c("pmid", "grasp_trait", "grasp_pvalue"), sep=",") %>%
               mutate_at(vars(grasp_pvalue), as.double) %>%
               group_by(rsid, pmid, grasp_trait, grasp_pvalue) %>%
               summarize(query_snp_rsids=toString(query_snp_rsid)) %>%
               ungroup() %>%
               arrange(grasp_pvalue)
       }
    )
}

# explore_haplo("R", "ALLonly", "DRM", "1y", "grasp") %>% 
#     arrange(grasp_pvalue) 
# 
# explore_haplo("R", "ALLonly", "DRM", "1y", "motifs") %>% 
#     print(n=35)
# explore_haplo("R", "ALLonly", "DRM", "1y", "proteins") %>% 
#     print(n=20)
# explore_haplo("R", "ALLonly", "DRM", "1y", "chromatin_marks")
# explore_haplo("R", "ALLonly", "DRM", "1y", "promoter")
# explore_haplo("R", "ALLonly", "DRM", "1y", "enhancer")

explore_haplo("R", "ALLonly", "DRM", "1y", "grasp") %>% 
    arrange(grasp_pvalue) %>%
    rename(PMID=pmid,
       `GRASP Trait`=grasp_trait,
       `GRASP P-value`=grasp_pvalue) %>%
    unite(`RSID/ Query SNP`, c("rsid", "query_snp_rsids"), sep="/ ") %>%
    knitr::kable(caption="\\label{tab:r_drm_grasp} GRASP hits for rs1051680", booktabs=TRUE, align='c') %>%
    kableExtra::kable_styling(latex_options="striped",
                              font_size=9,
                              position='center') %>%
    kableExtra::column_spec(1, width="5em") %>%
    kableExtra::column_spec(2, width="5em") %>%
    kableExtra::column_spec(3, width="15em") %>%
    kableExtra::column_spec(4, width="5em") 
# explore_haplo("R", "ALLonly", "DRM", "1y", "eqtl") %>% 
#     print(n=122)
```

## Exploratory analyses: Recipient SNP associations with Organ Failure 

\begin{figure}
    \centering
    \includegraphics[width=\linewidth, height=3.5in]{~/Desktop/figures/chapter5/r_of_1y_gwas_hit.png}
    \caption[Organ failure in recipient on chromosome 6.]{Organ failure in recipient on chromosome 6. Sentinel SNP is rs6915410 and colored blue. The region comprises +/- 150 kb window from the sentinel SNP. The red-yellow color gradient represents LD strength. Shapes represent different effects that the SNP has been reported to have in this region from independent studies.}
    \label{fig:r_of_1y}  
\end{figure}

```{r, echo=FALSE, eval=FALSE}
joint_res_split <- joint_res %>% 
    filter(PVALUE_M < 5e-06, PVALUE_c1 < 0.05 , PVALUE_c2 < 0.05) %>%
    mutate(pt_subset=str_replace(pt_subset, "ALLsubtype_Bcell", "BALL")) %>%  
    separate(pt_subset, c("genome", "disease", "outcome", "interval"), sep="_")

explore_haplo("R", "ALLonly", "OF", "1y", "proteins") %>%
    select(-n) %>% 
    mutate(protein=toupper(protein)) %>%
    group_by(protein) %>%
    summarize(cell_line=toString(cell_line)) %>%
    rename(`Protein Bound`=protein,
           `Cell Lines`=cell_line) %>%
    knitr::kable(caption="\\label{tab:r_of_proteins} Proteins with Binding Evidence in rs6915410 Region.", 
                 booktabs=TRUE,
                 align='c') %>%
    kableExtra::kable_styling(latex_options="striped",
                              font_size=9,
                              position='center') %>%
    kableExtra::column_spec(1, width="10em") %>%
    kableExtra::column_spec(2, width="30em") %>%
    footnote(general_title="",
         general="All SNPs in this region are minimum $r^2 \\\\geq 0.8$ with rs6915410.",
         footnote_as_chunk = TRUE,
         threeparttable = TRUE, 
         escape=FALSE)
```


```{r, echo=FALSE, eval=FALSE}
explore_haplo("R", "ALLonly", "OF", "1y", "motifs") %>%
    mutate(motifs=str_replace_all(motifs, "_[:alnum:]+", "")) %>% 
    select(-n) %>%
    count(motifs) %>%
    group_by(n) %>%
    mutate(motifs=toupper(motifs)) %>%
    summarize(motifs=toString(motifs)) %>%
    arrange(-n) %>%
    select(motifs, n) %>%
    rename(`Motifs`=motifs,
           `Number of SNPs`=n) %>%
    knitr::kable(caption="\\label{tab:r_of_motifs} Regulatory Motifs Altered in Recipient Organ Failure rs6915410 Region.", booktabs=TRUE, align='c') %>%
    kableExtra::kable_styling(latex_options="striped",
                              font_size=9,
                              position='center') %>%
    kableExtra::column_spec(1, width="25em") %>%
    kableExtra::column_spec(2, width="10em") %>%
    footnote(general_title="",
             general="All SNPs in this region are minimum $r^2 \\\\geq 0.8$ with rs6915410.",
             footnote_as_chunk = TRUE,
             threeparttable = TRUE, 
             escape=FALSE)
```    

While more recipients died due to transplant in the first year compared to death due to disease (Table \ref{tab:all_event_props}), no genome-wide associations with recipient SNPs were uncovered from TRM survival analyses (Manhattan Plot not shown). 

However, upon when we scanned suggestive associations organ failure in DISCOVeRY-BMT, we identified rs6915410 ($P = 4.17\times{10}^{-7}$; HR: 4.64 [2.56-8.38]; C effect allele) on chromosome 6 as an interesting locus for death by organ failure (OF) in recipients. This SNP is in a very active region, where many SNPs that are in moderate to high LD have direct regulatory effects on *DEF6* and *PPARD* (Figure \ref{fig:r_of_1y}). In addition to rs7744392 being *DEF6* eQTL in peripheral blood monocytes [@Zeller_2010] and blood [@Fehrmann_2011], it is an eQTL for *MPO* in peripheral monocytes [@Zeller_2010] as well. These eQTLs were identified in lymphoblastoid and whole blood derived cell lines [@Zeller_2010; @Fehrmann_2011]. Blood is the most abundant tissue that has histone marks on enhancers and promoter regions for rs6915410 and corresponding correlated SNPs [@haploreg]. Experimental ChIP-seq studies have shown that \textit{ER$\alpha$} protein binds to this region. Notably, this region has evidence of altered regulatory motifs including known leukemia genes *E2A*, *AR3A*, and *CEBPB*, *FOXO*, *GATA*, *STAT* families [@Iacobucci_2017]. 


## Post-HSCT Associations with Donor SNPs
All eight patient outcomes (DRM, TRM, OS, PFS, REL, GVHD, INF and OF) in ALL patients were tested for association with their matched unrelated donors SNPs. We detected genome-wide associated regions with OS, DRM, TRM and GVHD outcomes. Identifying such donor SNPs can be helpful in providing insight on genes and pathways involved in post-transplant events. More importantly upon rigorous replication, these SNPs can help clinicians improve donor selection and ultimately post-transplant survival outcomes.

## Donor SNPs Associated with Recipient Overall Survival

\begin{figure}
    \centering
    \includegraphics[width=\linewidth, height=3.5in]{~/Desktop/figures/chapter5/D_ALLonly_OS.jpeg}
    \caption[Manhattan Plot of Donor ALL OS GWAS.]{Manhattan Plot of Donor ALL OS GWAS. The x-axis is chromosome 1-22 and the y-axis is the $-\log_{10}(P_{meta})$. Each dot is a SNP. The red line is genome wide significance at $P_{meta} < 5\times{10}^{-8}$. The blue line is suggestive significance at $P < 5\times{10}^{-5}$. Labeled SNPs are associated hits that have passed genome-wide threshold.}
    \label{fig:d_os_1y}  
\end{figure}


```{r, echo=FALSE}
joint_res <- readRDS("~/Google Drive/OSU_PHD/DBMT_ALLonly/hits/20190306_suggestive_sig_snps_all_outcomes.rds")

joint_res %>%
    filter(pt_subset=="D_ALLonly_OS_1y",
           PVALUE_M < 5e-08,
           PVALUE_c1 < 5e-05,
           PVALUE_c2 < 0.2,
           HR_ADJ > (1/15) & HR_ADJ < 15) %>% 
    arrange(CHR, PVALUE_M) %>%
    unite(`CHR:POS`, c("CHR", "POS"), sep=":") %>% 
    mutate_at(.vars=vars(HR_ADJ, HR_lower_ADJ, HR_upper_ADJ), .funs=funs(round(., 2))) %>%
    mutate_at(.vars=vars(RefPanelAF, SAMP_MAF_c1, SAMP_MAF_c2), .funs=funs(round(., 4))) %>%
    mutate(INFO=round(INFO, 2),
           HR=glue_data(., "{HR_ADJ} ({HR_lower_ADJ}-{HR_upper_ADJ})")) %>%
    select(-PVALUE_c1:-PVALUE_c2, -Direction, -TYPED, -HR_ADJ:-regdb_score, INFO) %>%
    unite(`Ref. Panel/Cohort 1/ Cohort 2 MAF`, c("RefPanelAF", "SAMP_MAF_c1", "SAMP_MAF_c2"), sep="/") %>%
    rename(`HR (95% CI)`=HR,
           `REF/ EFF`=REF_ALT, 
           `P-value`=PVALUE_M) %>%  
        kable(caption="\\label{tab:d_os_hits} DISCOVeRY-BMT Donor SNPs Associated to Recipient Overall Survival at 1-Year", 
              digits=60, 
              align='c') %>% 
    kableExtra::kable_styling(latex_options="striped",
                              font_size=9, 
                              position='center') %>%
    kableExtra::column_spec(1, width="5em") %>%
    kableExtra::column_spec(2, width="5.5em") %>%
    kableExtra::column_spec(3, width="2em") %>%
    kableExtra::column_spec(4, width="10.5em") %>%
    kableExtra::column_spec(5, width="4.8em") %>%
    kableExtra::column_spec(6, width="8em") %>%
    footnote(general_title="",
             general="All SNPs shown here are imputed. Genomic positions are GRCh37 reference genome. REF is reference allele and EFF is effect allele.",
             footnote_as_chunk = TRUE,
             threeparttable = TRUE)
```


\begin{figure}
    \centering
    \includegraphics[width=\linewidth, height=3.5in]{~/Desktop/figures/chapter5/rs62465751_7_D_ALLonly_OS.png}
    \caption[Donor OS Region Chromosome 7.]{Donor OS Region Chromosome 7. Sentinel SNP is rs62465751 and colored blue. The region comprises +/- 150 kb window from the sentinel SNP. The red-yellow color gradient represents LD strength. Shapes represent different effects that the SNP has been reported to have in this region from independent studies.}
    \label{fig:d_os_chr7}  
\end{figure}

Donor SNPs were associated to recipient overall survival (OS) on chromosomes 2, 5, 9, 13, and 18 (Figure \ref{fig:d_os_1y} and Table \ref{tab:d_os_hits}).

There are two associated regions on chromosome 2 (near 71Mb and 154 Mb). The former has two associated SNPs in perfectly correlated (rs116220248, rs76480148), both of these SNPs are in perfect LD with a SNP that has protein binding evidence for CEBPB, a well known leukemia target gene [@Akasaka_2007;@Guerzoni_2006]. The other region ($\approx$ 154 Mb region) does not have any noteworthy functional effects. Similarly, the regions on 5, 9, 13, and 18 also provided little evidence of having any known eQTLs, promoters Histones, or relevant alterations to cancer related motifs [@haploreg; @snipa].

Upon screening suggestive associations of donor SNPs with patient OS, we detected rs62465751, a highly regulatory SNP located on chromosome 7, upstream of *TMEM248* (Figure \ref{fig:d_os_chr7}). This location encompassing rs62465751 has been predicted to be an active transcription start site in almost all cell lines included in Roadmap Epigenome project. Consequently, rs62465751 was detected for 60 eQTL associations, involving multiple genes across multiple tissues included in GTEx project. rs62465751 belongs to a LD block consisting of 73 SNPs ($r^2>0.8$), 22 of which located in the intronic region of *TMEM248*. SNPs in this LD block alter more than 22 binding motifs and are all reported eQTLs for various tissues, including CEU ancestry lymphoblastoid cell lines [@stranger_2007; @Lappalainen_2013]. 

<!-- ## DISCOVeRY-BMT Associations have been previously reported in independent GWAS  -->
<!-- ```{r gwas_hits_1y, echo=FALSE, eval=FALSE} -->
<!-- gwas_hits <- readRDS("~/Google Drive/OSU_PHD/DBMT_ALLonly/hits/20190228_gwas_hits.rds") -->
<!-- gwas_hits %>% -->
<!--     filter(str_detect(analysis, "1y")) %>% -->
<!--     separate(analysis, c("genome", "disease", "outcome", "interval"), sep="_") %>% -->
<!--     select(-interval, -disease, -query_snp_rsid) %>% -->
<!--     mutate(genome=ifelse(genome=="D", "Donor", "Recipent")) %>% -->
<!--     unite(analysis, c("genome", "outcome"), sep="\n") %>% -->
<!--     rename(RSID=rsid, -->
<!--            PMID=pmid, -->
<!--            `GWAS Trait`=gwas_trait, -->
<!--            `GWAS P-value`=gwas_pvalue, -->
<!--            Analysis=analysis) %>% -->
<!--     select(Analysis, everything()) %>% -->
<!--     kable(caption="\\label{tab:gwas_hits_1y} Previously Published GWAS Traits from GWAS Catalog Reproduced in DISCOVeRY-BMT 1 Year ALL Analysis.", digits=60, align='c') %>% -->
<!--     kableExtra::kable_styling(latex_options="striped", font_size=9, position='center') %>% -->
<!--     kableExtra::column_spec(1, width="4em") %>% -->
<!--     kableExtra::column_spec(2, width="5em") %>% -->
<!--     kableExtra::column_spec(3, width="4em") %>% -->
<!--     kableExtra::column_spec(4, width="12.2em") %>% -->
<!--     kableExtra::column_spec(5, width="5em") %>% -->
<!--     footnote(general_title="", -->
<!--              general="SNPs shown here are either the query SNP or in LD ($r^{2}$ > 0.8) with the query SNP.", -->
<!--              footnote_as_chunk = TRUE, -->
<!--              threeparttable = TRUE,  -->
<!--              escape=FALSE) -->
<!-- ``` -->



<!-- Table \ref{tab:gwas_hits_1y} yielding four regions that had previously been reported upon. While all of these previously identified traits are not blood or cancer related, inspecting these regions in DISCOVeRY-BMT is warranted. -->


## Donor SNPs Associated with Recipient Disease Related Mortality 

\begin{figure}
    \centering
    \includegraphics[width=\linewidth, height=3.5in]{~/Desktop/figures/chapter5/D_ALLonly_DRM.jpeg}
    \caption[Manhattan Plot of Donor SNPs Contribution to Recipient DRM. ]{Manhattan Plot of Donor SNPs Contribution to Recipient DRM. The x-axis is chromosome 1-22 and the y-axis is the $-\log_{10}(P_{meta})$. Each dot is a SNP. The red line is genome wide significance at $P_{meta} < 5\times{10}^{-8}$. The blue line is suggestive significance at $P_{meta} <  5\times{10}^{-5}$. Labeled SNPs are associated hits that have passed genome-wide threshold.}
    \label{fig:d_drm_1y}  
\end{figure}

```{r, echo=FALSE}
joint_res <- readRDS("~/Google Drive/OSU_PHD/DBMT_ALLonly/hits/20190306_suggestive_sig_snps_all_outcomes.rds")
joint_res %>%
    filter(pt_subset=="D_ALLonly_DRM_1y", PVALUE_M < 5e-08, PVALUE_c1 < 5e-05,
           PVALUE_c2 < 0.2, HR_ADJ > (1/15) & HR_ADJ < 15) %>% 
    arrange(CHR, PVALUE_M) %>%
    unite(`CHR:POS`, c("CHR", "POS"), sep=":") %>% 
    mutate_at(.vars=vars(HR_ADJ, HR_lower_ADJ, HR_upper_ADJ), .funs=funs(round(., 2))) %>%
    mutate_at(.vars=vars(RefPanelAF, SAMP_MAF_c1, SAMP_MAF_c2), .funs=funs(round(., 4))) %>%
    mutate(INFO=round(INFO, 2),
           HR=glue_data(., "{HR_ADJ} ({HR_lower_ADJ}-{HR_upper_ADJ})")) %>%
    select(-PVALUE_c1:-PVALUE_c2, -Direction, -TYPED, -HR_ADJ:-regdb_score, INFO) %>%
    unite(`Ref. Panel/Cohort 1/ Cohort 2 MAF`, c("RefPanelAF", "SAMP_MAF_c1", "SAMP_MAF_c2"), sep="/") %>%
    rename(`HR (95% CI)`=HR,
           `REF/ EFF`=REF_ALT, 
           `P-value`=PVALUE_M) %>%  
    kable(caption="\\label{tab:d_drm_hits} DISCOVeRY-BMT Donor Genotypes Associated with Recipient Disease Related Mortality at 1-year.", 
              digits=60, 
              align='c') %>% 
    kableExtra::kable_styling(latex_options="striped",
                              font_size=9, 
                              position='center') %>%
    kableExtra::column_spec(1, width="5em") %>%
    kableExtra::column_spec(2, width="5.5em") %>%
    kableExtra::column_spec(3, width="2em") %>%
    kableExtra::column_spec(4, width="10.5em") %>%
    kableExtra::column_spec(5, width="4.8em") %>%
    kableExtra::column_spec(6, width="8em") %>%
    footnote(general_title="",
             general="All SNPs shown are imputed. Genomic positions are GRCh37 reference genome. REF is reference allele and EFF is effect allele.",
             footnote_as_chunk = TRUE,
             threeparttable = TRUE)
```

We found 5 regions associated with increased risk for DRM when donor SNPs had the risk allele in single regions on 17, 20, and 21, and two regions on chromosomes 2 and 5 (Figure \ref{fig:d_drm_1y};Table \ref{tab:d_drm_hits}). 

On chromsome 2, rs75569504 (Table \ref{tab:d_drm_hits}) is in strong LD with SNPs that are intronic for *MYO3B*. This marker is in perfect LD with SNPs that bind proteins TFs Setdb1 and Hae2f1, however, little evidence exists relating these TFs to ALL. The other region on chromosome 2, rs79503405 was associated with increased risk in ALL DRM and in perfect LD with an eQTL (in testes cell lines) a synonymous variant on *LRP2* (rs35114151). Synonymous mutations may affect DNA replication processes such as transcription, splicing or translation, any of which can have implications on phenotype.  

Chromosome 5 marker rs72807345 is intronic for *SQSTM1*, a gene that has shown to be involved in somatic chromosomal abberations in adult T-cell leukemias [@gorello_2010]. The markers are on 17 is intronic for *ASIC2* and binds proteins *GR* (a highly targetable nuclear receptor and associated with T-ALL resistance) [@McMaster_2008; @Piovan_2013] and *STAT3* (a well known leukemia target gene) [@Adamaki_2015;@Adamaki_2015]. When looking at the regional association plots for the aforementioned DRM associations, SNiPA database did not reveal any notable effects on transcriptional regulation [@snipa]. 

The regions on chromosomes 20 and 21 showed no evidence of affecting transcriptional processes from functional studies [@haploreg; @snipa].  

## Donor SNPs Associated with Recipient Transplant Related Mortality

\begin{figure}
    \centering
    \includegraphics[width=\linewidth, height=3.5in]{~/Desktop/figures/chapter5/D_ALLonly_TRM.jpeg}
    \caption[Manhattan Plot of Donor SNPs Contribution to Recipient TRM]{Manhattan Plot of Donor SNPs Contribution to Recipient TRM. The x-axis is chromosome 1-22 and the y-axis is the $-\log_{10}(P_{meta})$. Each dot is a SNP. The red line is genome wide significance at $P_{meta} < 5\times{10}^{-8}$. The blue line is suggestive significance at $P_{meta} <  5\times{10}^{-5}$. Labeled SNPs are associated hits that have passed genome-wide threshold.}
    \label{fig:d_trm_1y}  
\end{figure}

```{r, echo=FALSE}
joint_res %>%
    filter(pt_subset=="D_ALLonly_TRM_1y",
           PVALUE_M < 5e-08,
           PVALUE_c1 < 5e-05,
           PVALUE_c2 < 0.2,
           HR_ADJ > (1/15) & HR_ADJ < 15) %>% 
    arrange(CHR, PVALUE_M) %>%
    unite(`CHR:POS`, c("CHR", "POS"), sep=":") %>% 
    mutate_at(.vars=vars(HR_ADJ, HR_lower_ADJ, HR_upper_ADJ), .funs=funs(round(., 2))) %>%
    mutate_at(.vars=vars(RefPanelAF, SAMP_MAF_c1, SAMP_MAF_c2), .funs=funs(round(., 4))) %>%
    mutate(INFO=round(INFO, 2),
           HR=glue_data(., "{HR_ADJ} ({HR_lower_ADJ}-{HR_upper_ADJ})")) %>%
    select(-PVALUE_c1:-PVALUE_c2, -Direction, -TYPED, -HR_ADJ:-regdb_score, INFO) %>%
    unite(`Ref. Panel/Cohort 1/ Cohort 2 MAF`, c("RefPanelAF", "SAMP_MAF_c1", "SAMP_MAF_c2"), sep="/") %>%
    rename(`HR (95% CI)`=HR,
           `REF/ EFF`=REF_ALT, 
           `P-value`=PVALUE_M) %>%  
    kable(caption="\\label{tab:r_os_hits} DISCOVeRY-BMT Donor Genotypes Associated with Recipent Transplant Related Mortality at 1-year.", 
              digits=60, 
              align='c') %>%
    kableExtra::kable_styling(latex_options="striped",
                              font_size=9, 
                              position='center') %>%
    kableExtra::column_spec(1, width="5em") %>%
    kableExtra::column_spec(2, width="5.5em") %>%
    kableExtra::column_spec(3, width="2em") %>%
    kableExtra::column_spec(4, width="10.5em") %>%
    kableExtra::column_spec(5, width="4.8em") %>%
    kableExtra::column_spec(6, width="8em") %>%
    footnote(general_title="",
             general="All SNPs shown are imputed. Genomic positions are GRCh37 reference genome. REF is reference allele and EFF is effect allele.",
             footnote_as_chunk = TRUE,
             threeparttable = TRUE)
```


Our analyses revealed TRM associations with donor SNPs on chromosomes 2, 3, 15, 18 and 20 (Table \ref{tab:r_os_hits}; Figure \ref{fig:d_trm_1y}). The associated SNPs in chromosome 2 are in LD and are located in the intronic region of *MYO3B* gene [@haploreg]. The chromosome 2 associations are driven by disease death, as we also observed significant SNPs in this region for DRM. Although lack of *FHIT* at the protein level has been shown to induce leukemia [@peters_1999], the associated intronic SNP on chromosome 3 had minimal evidence of transcriptional regulation from existing literature based on experimental evidence [@snipa]. Similarly, the associated chromosome 18 region for donor SNPs showed effect on transcription [@haploreg; @snipa]. 


<!-- \begin{figure} -->
<!--     \centering -->
<!--     \includegraphics[width=\linewidth, height=3.5in]{~/Desktop/figures/chapter5/rs75433789_2_D_ALLonly_TRM.png} -->
<!--     \caption[Regional association plot of recipient SNP rs75433789 with DRM in ALL patients.]{Regional association plot of recipient SNP rs75433789 with DRM in ALL patients. The x-axis is genomic position on chromosome 2 and the y-axis is the $-\log_{10}(P_{meta})$. Sentinel SNP is rs75433789 and colored blue. The region comprises +/- 150 kb window from the sentinel SNP. The red-yellow color gradient represents LD strength. Shapes represent various regulatory or functional effects that the SNP has been reported to have in this region from independent studies.} -->
<!--     \label{fig:d_trm_1y_chr2reg}   -->
<!-- \end{figure} -->


```{r, echo=FALSE, eval=FALSE}
joint_res_split <- joint_res %>% 
    filter(PVALUE_M < 5e-06, PVALUE_c1 < 0.05 , PVALUE_c2 < 0.05, RSID=="rs9783053") %>%
    mutate(pt_subset=str_replace(pt_subset, "ALLsubtype_Bcell", "BALL")) %>%  
    separate(pt_subset, c("donor", "disease", "outcome", "interval"), sep="_")

explore_haplo("D", "ALLonly", "GVHD", "1y", "motifs") %>%
    mutate(motifs=str_replace_all(motifs, "_[:alnum:]+", "")) %>% 
    select(-n) %>%
    count(motifs) %>%
    group_by(n) %>%
    mutate(motifs=toupper(motifs)) %>%
    summarize(motifs=toString(motifs)) %>%
    arrange(-n) %>%
    select(motifs, n) %>%
    rename(`Motifs`=motifs,
           `Number of SNPs`=n) %>%
    knitr::kable(caption="\\label{tab:d_gvhd_motifs} Regulatory Motifs Altered in Recipient Organ Failure rs6915410 Region.", 
                 booktabs=TRUE, 
                 align='c',
                 digits=20) %>%
    kableExtra::kable_styling(latex_options="striped",
                              font_size=9,
                              position='center') %>%
    kableExtra::column_spec(1, width="30em") %>%
    kableExtra::column_spec(2, width="10em") %>%
    footnote(general_title="",
             general="All SNPs in this region are minimum $r^2 \\\\geq 0.8$ with rs9783053",
             footnote_as_chunk = TRUE,
             threeparttable = TRUE, 
             escape=FALSE)
```


## Donor SNPs Associated with Graft-versus-Host Disease

\begin{figure}
    \centering
    \includegraphics[width=\linewidth, height=3.5in]{~/Desktop/figures/chapter5/rs9783053_1_D_ALLonly_GVHD.png}
    \caption[Donor GVHD Region Chromosome 1.]{Donor GVHD Region Chromosome 1. Sentinel SNP is rs9783053 and colored blue. The region comprises +/- 150 kb window from the sentinel SNP. The red-yellow color gradient represents LD strength. Shapes represent different effects that the SNP has been reported to have in this region from independent studies.}
    \label{fig:d_gvhd_1y}  
\end{figure}


```{r, echo=FALSE, eval=FALSE}
explore_haplo("D", "ALLonly", "GVHD", "1y", "eqtl") %>% 
    filter(!str_detect(gene, "ensg")) %>%
    select(-n) %>%
    mutate(gene=toupper(gene)) %>%
    group_by(gene) %>%
    summarize(cell_lines=toString(cell_lines)) %>%
    rename(Gene=gene,
           `Cell Lines`=cell_lines) %>%
        knitr::kable(caption="\\label{tab:d_gvhd_eqtl} eQTLs in Donor GVHD rs9783053.", booktabs=TRUE, align='c', digits=20) %>%
    kableExtra::kable_styling(latex_options="striped",
                              font_size=9,
                              position='center') %>%
    kableExtra::column_spec(1, width="5em") %>%
    kableExtra::column_spec(2, width="15em") %>%
    footnote(general_title="",
             general="All SNPs in this region are minimum $r^2 \\\\geq 0.8$ with rs9783053.",
             footnote_as_chunk = TRUE,
             threeparttable = TRUE, 
             escape=FALSE)
```

```{r, echo=FALSE, eval=FALSE}
explore_haplo("D", "ALLonly", "GVHD", "1y", "grasp") %>% 
    arrange(grasp_pvalue) %>%
    rename(PMID=pmid,
       `GRASP Trait`=grasp_trait,
       `GRASP P-value`=grasp_pvalue) %>%
    unite(`RSID/ Query SNP`, c("rsid", "query_snp_rsids"), sep="/ ")  %>%
    knitr::kable(caption="\\label{tab:d_gvhd_grasp} GRASP hits for rs9783053.", booktabs=TRUE, align='c', digits = 20) %>%
    kableExtra::kable_styling(latex_options="striped",
                              font_size=9,
                              position='center') %>%
    kableExtra::column_spec(1, width="5em") %>%
    kableExtra::column_spec(2, width="5em") %>%
    kableExtra::column_spec(3, width="15em") %>%
    kableExtra::column_spec(4, width="5em") 

```

We scanned suggestive SNP regions ($P < 5\times{10}^{-5}$) for regions that may affect transcription factor binding. Donor GVHD SNP rs9783053 (typed SNP; $P=1.41\times{10}^{-6}$; HR: 3.27 [2.02-5.29]; G effect allele) was found to have a regulomeDB score of 1A (the highest score). This means based off of previous literature, there is strong evidence that rs9783053 affects transcription factor binding [@Boyle_2012]. This region has many SNPs that directly regulate transcription on *UBE4B* (Figure \ref{fig:d_gvhd_1y}). This polymorphism has five proteins (Ctcf, Ebf1, Elf1, Nf$\kappa$b, Pu1) that bind to this SNP in lymphoblastoid cell lines [@encode_2012]. This variant is in LD with synonymous variant rs2273299 ($r^2=0.88$). Experimental cell line studies from ENCODE provide evidence suggesting that this region alters regulatory motifs for cancer and leukemia related genes [@haploreg]. Variants that alter gene expression were found in blood and lymphoblast derived CEU cell lines. Thus, while donor GVHD SNPs did not reach genome-wide significance, the is some evidence that suggests this region may transcriptional or epigenetic on effects on ALL related genes.

# Discussion

Previously the relationship between survival outcomes and non-HLA donor-recipient genetics after MUD-HSCT had not been studied at the genome-wide level. Here we presented the first GWAS investigating this objective. The cohorts of DISCOVeRY-BMT span the entire ALL patient pool, ranging from childhood, AYA, adults and senior adults. Genome-wide associations to survival outcomes after 1-year post-transplant were identified in both recipient and donor genomes. Specifically, recipient DRM, OS, and OF. And for donors, associations were found for DRM, TRM, GVHD, and OS. Often our genome-wide associations provided only a small insights of possible biological mechanisms that may contribute to these observed death related phenotypes. 

We attempted to gain additional insight by leveraging publicly available databases. We inspected regions that were below genome-wide significance at $P < 5\times{10}^{-5}$. Often we found regions that were in fact not genome-wide significant, but regional association plots revealed that there may some effect on transcription and that previous studies have provided evidence that these regions have many interactions. Future directions for this project include implementing PrediXcan which is currently underway in the lab and will be added to this work before publication. PrediXcan always assessment of gene-based expression levels in our data and thus will help us prioritize our GWAS associations. Future directions for this project include implementing PrediXcan [@Barbeira_2018] which is currently underway in the lab and will be added to this work before publication. PrediXcan allows assessment of gene-based expression levels in our data (using blood tissue samples from GTEx) and thus will help us prioritize our GWAS associations.

Although DISCOVeRY-BMT is the largest cohort to date investigating ALL recipient-donor genetic variation after transplant, sample size remains a large limitation. The next step will be replicate these findings in an independent study. Namely using University of Washington ALL cohort study for replication of relapse and overall survival. If available, in addition to replication using University of Washington cohorts, meta-analysis combining DISCOVeRY-BMT may provide stronger signal for overall survival and/or relapse outcomes. In terms of improving our survival models, alterations to our current workflow may improve signal by adjusting for age by stratifying by clinical age groups, or by modeling the age as a non-linear function and smoothing the distribution using splines [@shepherd_2017]. Challenges in improving long-term survival after MUD-HSCT remain, however, we believe that this study provides unique insights in this field and is a positive step towards improving outcomes. 






