---
title: "introduction"
author: "Abbas Rizvi"
date: "9/14/2018"
output: pdf_document  
---

```{r ch1_setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Broadly, this dissertation examines genetic variation in the context hematopoetic stem cell transplantation (HSCT). This dissertation seeks to identify and characterize genetic variants in patients who have acute myeloid leukemia (AML), acute lymphoblastic leukmia (ALL), or myelodysplastic syndrome (MDS) and received an HSCT from an HLA matched-unrelated donor (MUD). This dissertation also seeks to enhance the computational workflows that are used in these fields. The significance is three-fold, first, we can identify clinically relevant markers that may improve donor selection beyond traditional methods; second, we can characterize pre-transplant risk of disease or transplant related death within the first year; and third, we can facilitate other researchs studying similar problems and data by developing open-source software. 

The project that this dissertation contributes to has the opportunity to be a real life example of a translational study (from computational analysis of biological data to bedside). The dissertation is broken down such that the introduction introduces genetic assocation studies and important related concepts (genetic and statistical) that are needed to appreciate and understand the underlying analysis. It then introduces the DISCOVeRY-BMT GWAS and clinical data in detail. Chapter 2 will discuss the literature review and replication/validation study that we published on. Chapter 3 discusses the R package that we developed and details the process of developing that package. Chapter 4 is the application of the R package and custom pipeline developed to perform large automated GWAS. Chapter 5 discusses the discovery and inference of markers in ALL donor and recipient pairs. This dissertation ends with Chapter 6, which begins undercovering genetic contributions to early death (the first 100 days) after transplant and the future directions this question should undergo. 


# Genetic Association Studies
Human genomes can differ from one another at single genomic positions such as single nucleotide polymorphisms (SNPs), or genomes may have larger structural changes, such as microsatellites regions, insertions/deletions (indels), copy number variations (CNVs), or variable-tandem repeats (VNTRs). To further understand genetic architecture as it relates to disease risk or to physical quantitative traits (i.e. height, weight), gene mapping studies are used to determine associations between genomic DNA sequence variations and phenotypic variability. These association mapping studies have increasingly evolved from looking at specific regions of interest (candidate gene assocation studies) to more agnostic approaches that investigate larger portions of the genome, such as genome wide assocation studies (GWAS), whole-exome sequencing studies (WES studies) and whole-genome sequencing studies (WGS studies). This dissertation is primarily focused on GWAS, specifically the application of using GWAS in the context of identifying common variants in after hematopoetic stem cell transplantation (HSCT), where more details will be discussed in subsequent subections of the Introduction. 

GWAS employ genome-wide genotyping microarrays to measure genetic variation -- and they have become the standard platform in academic and industry to test for association of phenotype with common genetic variants. Common genetic variants are defined as those with a *minor allele frequency* (MAF) of $\geq$ 1% and rare variants are defined as those with a MAF $\leq$ than 1%. Genotyping microarrays can be designed to contain rare variants. GWAS ask if the allele of a genetic variant is found more often than what would be expected than by random chance in individuals with the phenotype of interest. (e.g. the disease being studied). 

Many novel assocations that are identified are in non-coding regions 

Genetic association studies aim to test for association between one or more single nucleotide polymorphisms (SNPs) and phenotype (trait), that may be attributable to disease or some other quantitative (i.e. height or weight) or discrete characteristic. 

Individuals with higher frequencies of SNPs or genotypes may be interpreted to have increased risk of a specific phenotype. SNPs are the most widely used genetic markers. 


Genetic associations studies usually fall under two main subcategories, candidate gene association studies (CGAS) or genome wide association studies (GWAS). CGAS are assocation studies that investigate specific genes or regions of interest. Typically these are performed when researchers believe that the underlying biology is understood and they want to identify specific markers that contribute to genetic variation in these 'known' regions. 

GWAS are observational studies that aim to identify genetic variation 


## Linkage Disequilibrium


## Genetic Imputation


# DISCOVeRY-BMT

\underline{D}etermining the \underline{I}nfluence of \underline{S}usceptibility \underline{CO}nveying \underline{V}ariants \underline{R}elated to one-\underline{Y}ear mortality after \underline{B}one \underline{M}arrow \underline{T}ransplant (DISCOVeRY-BMT) is a GWAS. This GWAS aims at identifying and characterizing non-human leukocyte antigen (HLA) genetic variation in the context of survival outcomes in acute lymphoblastic leukemia (ALL), acute myelogenous leukemia (AML) and myleodastic dysplastic syndrome (MDS) patients and their matched unrelated donor after hematopoietic stem cell transplantation (HSCT). HSCT is also called bone marrow transplantation (BMT) and hence both terms will be used interchangeably throughout this document. 

## Hematopoetic Stem Cell Transplantation
HSCT is an established therapeutic procedure that is used as a potentially curative treatment for life-threatening congenital or acquired blood disorders (malignant or non-malignant) [@Henig_2014]. HSCT involves the intravenous infusion of autologous or allogeneic hematopoietic progenitor cells to restore normal function in patients whose bone marrow is compromised. Autologous HSCT involves self-donation of marrow stem cells, whereas allogeneic HSCT is when stem cells are transferred from a HLA-matched related donor (MRD) or a HLA-matched unrelated donor (MUD). Although a matched sibling donor is preferred, only approximately 30% of patients who may benefit from HSCT have such a donor available. In the United States, the number of allogeneic transplants yearly has dramatically risen over the past decade, across all diseases [@Pasquini_2013]. Patients with acute myeloid leukemia (AML), acute lymphoblastic leukemia (ALL), or myelodysplastic syndrome (MDS), represent the largest group treated with allogeneic HSCT. While both patient care and matching has improved over the past few decades almost half of all high-resolution 10/10 MUD-HSCT recipients die within one-year post-transplant due to either their disease or transplant-related causes [@Pasquini_2013]. These trends also show transplant-related causes are a larger contributor to mortality within the first 100-days post-transplant and shift towards primary disease after approximately six months post-transplant [@DSouza_2017]. Reducing TRM without increasing risk of disease death and vice versa continue to represent a substantial clinical challenge.

## Genotyping and Quality Control
All samples were genotyped using the Illumina Human OmniExpress BeadChip and the Illumina HumanExome BeadChip (University of Southern California Genomics Facility). 

Samples were assigned to plates to ensure the even distribution of patient characteristics and potential confounding variables using Optimal Sample Assignment Tool (OSAT), an R/Bioconductor software package [@OSAT].

Over 90% of DISCOVeRY-BMT patients self-reported as European American, Caucasian or White and thus replication and validation analyses are performed on these recipient-donor pairs. 


**Define population stratification**

Stringent quality control was performed on both samples and SNPs within this population. Population outliers were removed using EIGENSTRAT [@Price_AL] (n=73). 


Additional sample quality control removed samples with missing call rate â‰¥ 2% (n=54), sex mismatch (n=9), abnormal inbreeding coefficients (n=20), and evidence of cryptic relatedness (n=17), yielding 2106 and 777 donor-recipient pairs in cohorts 1 and 2, respectively. Typed SNPs were removed if the call rate was <98%, there was deviation from Hardy-Weinberg equilibrium proportions or discordance between duplicate samples was >2%. 

In total 637,655 and 632,823 SNPs from the OmniExpress BeadChip were available for imputation in cohorts 1 and cohort 2, respectively, using 1000 Genomes Project Phase 3 [@1000genomes]. 

IMPUTE2 [@Howie_2009] software was used for Imputation and QCTOOL was used to remove imputed genotypes with info score <0.7, certainty <0.7 and a minor allele frequency <0.005. The recipient-donor mismatch genome dosage calculations, described above, were done as the absolute value of the recipient minus donor minor allele dosages. Rs2066847 (SNP13) in NOD2/CARD15 was the only variant analyzed from the Illumina HumanExome as it was not typed on the OmniExpress chip or available following imputation.

## Patient Characteristics
DISCOVeRY-BMT investigates donor and recipient genetic factors that contribute to 1 year cause-specific mortality after MUD-HSCT [@Hahn_2015, @Clay_2017]. This GWAS comprises two independent cohorts. All patients in DISCOVeRY-BMT were reported to the Center for International Blood and Marrow Research (CIBMTR) and had banked biorepository samples from recipient and donor. The biorepository samples were genotyped and are the genetic information that is reported throughout this thesis. The genotyping will be discussed


cohort 1 included 2110 HSCT recipients and 2052 10/10 HLA matched unrelated donors from 2000 to 2008 (**Table 1.1**). These patients either had AML, ALL, or MDS. Prior to MUD-HSCT, the recipients were T cell-replete and treated with myeloablative (MA) or reduced intensity (RIC) conditioning regimens. 


In preparation for this GWAS, all cause-specific deaths were reviewed and judged by a consensus panel for for patients in DISCOVeRY-BMT were adjudicated post-mortem by an expert panel of clinicians over a 3-day period [@Hahn_2015]. The panel consisted of three HSCT oncologists and a HSCT clinical epidemiologist [@Hahn_2015]. 



```{r, echo=FALSE}
covariate.file <- read_tsv("~/Google Drive/OSU_PHD/DBMT_100d/DBMT_PhenoData_EA_long_allVar_20181206.txt")

covariate.file %>%
    group_by(sample_type, cohort) %>%
    tally() %>%
    ungroup() %>%
    mutate(cohort=ifelse(cohort==1, "Cohort 1: N (Percent %)", "Cohort 2: N (Percent %)")) %>% 
    arrange(sample_type) %>%  
    group_by(sample_type) %>%
    mutate(per=paste0("(", as.character(round((n/sum(n)*100),1)), "%)")) %>%
    ungroup() %>% 
    unite(n_per, c(n, per), sep=" ") %>%  
    spread(cohort, n_per) %>%
    rename(Genome=sample_type) %>%
    mutate(Genome=ifelse(Genome=="donor", "Donor", "Recipient")) %>%
    kable(format = "latex",
          caption="Donor and Recipients Cohorts in DISCOVeRY-BMT.", 
          booktabs=TRUE) %>%
    kable_styling(latex_options = "striped",
                  full_width=FALSE) %>%
    footnote(general_title="",
         general="Shown here are number of patients and donors in each cohort",
         footnote_as_chunk = TRUE,
         threeparttable = TRUE)

```


Disease Distribution
```{r, echo=FALSE}
dz.dist <- covariate.file %>%
    group_by(sample_type, cohort, disease) %>%
    tally() %>%
    ungroup() %>%
    mutate(cohort=ifelse(cohort==1, "cohort 1", "cohort 2")) %>% 
    arrange(cohort) %>% 
    gather(key, value, n) %>%
    unite(sample_cohort, c("sample_type", "cohort")) %>%
    spread(sample_cohort, value) %>%
    select(-key) %>%
    mutate_at(.vars=vars(contains("cohort")), .funs=funs(`per`=./sum(.))) %>%
    mutate(`donor_cohort 1`=paste0(`donor_cohort 1`, " (", round(`donor_cohort 1_per`, 2)*100, "%)"),
           `donor_cohort 2`=paste0(`donor_cohort 2`, " (", round(`donor_cohort 2_per`, 2)*100, "%)"),
           `recipient_cohort 1`=paste0(`recipient_cohort 1`, " (", round(`recipient_cohort 1_per`, 2)*100, "%)"),
           `recipient_cohort 2`=paste0(`recipient_cohort 2`, " (", round(`recipient_cohort 2_per`, 2)*100, "%)")) %>%
    select(-`donor_cohort 1_per`:-`recipient_cohort 2_per`) %>%
    as.data.frame()

rownames(dz.dist) <- dz.dist$disease
dz.dist$disease <- NULL

dz.dist %>%
    kable(format = "latex",
          caption="Donor and Recipients Disease Characteristics in DISCOVeRY-BMT.", 
          booktabs=TRUE) %>%
    kable_styling(latex_options = "striped",
                  full_width=FALSE) %>%
    add_header_above(c(" ", "Donor (N/Percent %)"=2, "Recipient (N/Percent %)"=2)) %>%
    footnote(general_title="",
             general="Shown here are disease distributions of ALL, AML and MDS in DISCOVeRY-BMT cohorts. The percentage in each cell is computed column-wise as the proportion of sample size (N) within a cohort across disease group.",
             footnote_as_chunk = TRUE,
             threeparttable = TRUE)
```



The primary cause of death was 


Defintions of outcomes:  
**Overall Survival (OS)**  
**Disease Related Mortality (DRM)**  
**Transplant Related Mortality (TRM)**
**TRM subtypes**

**Progression Free Survival (PFS)**  
**Relapse (REL)**  

Proportions of Events Mixed disease with outcomes

```{r, echo=FALSE}
mixed.outcome <- covariate.file %>%
    filter(disease %in% c("ALL", "AML", "MDS")) %>%
    group_by(sample_type, cohort) %>%
    summarize(OS=sum(dead_1Y),
              DRM=sum(disease_death_1Y),
              TRM=sum(TRM_1Y),
              PFS=sum(lfs_1Y),
              REL=sum(rel_1Y, na.rm=TRUE),
              GVHD=sum(GVHD_death_1Y),
              OF=sum(OF_1Y),
              INF=sum(infection_1Y)) %>% 
    arrange(cohort) %>%
    ungroup() %>%
    mutate(total=c(2052, 2110, 763, 777)) %>%
    mutate_at(.vars=vars(OS, DRM, PFS, REL, TRM, GVHD, OF, INF), .funs=funs(paste0(., " (", round((./total)*100, 2), "%)"))) %>%
    select(-total) %>%
    arrange(sample_type) %>%
    unite(sample_cohort, c("sample_type", "cohort")) %>% 
    gather(key, value, -sample_cohort) %>%
    spread(sample_cohort, value) %>% 
    rename(`Donor Cohort 1`=donor_1,
           `Donor Cohort 2`=donor_2,
           `Recipient Cohort 1`=recipient_1,
           `Recipient Cohort 2`=recipient_2) %>%
    mutate(key=factor(key, levels=c("OS", "DRM", "PFS", "REL", "TRM", "GVHD", "INF", "OF"))) %>%
    arrange(key) %>%
    as.data.frame()

rownames(mixed.outcome) <- mixed.outcome$key
mixed.outcome <- mixed.outcome %>%
    select(-key) 

mixed.outcome %>%
    kable(format = "latex",
          caption="Mixed Disease: Proportion of Events by Outcome and Cohort in Donor and Recipients", 
          booktabs=TRUE) %>%
    kable_styling(latex_options = "striped",
                  full_width=FALSE) %>%
    add_header_above(c(" ", "Donor (N/Percent %)"=2, "Recipient (N/Percent %)"=2)) %>%
    footnote(general_title="",
             general="Shown here are the proprtion of events for mixed disease (ALL, AML, and MDS) in both DISCOVeRY-BMT cohorts",
             footnote_as_chunk = TRUE,
             threeparttable = TRUE)
``` 

Explain that discrepancies between TRM and TRM subtypes are because some patients adjudication was not well differentiated between these subtypes.  

```{r, echo=FALSE, fig.cap="Histogram of Donor and Recipient Age Distributions in Mixed Disease (AML, ALL, and MDS)"}
covariate.file %>%
    select(disease, sample_type, cohort, age, dnrage) %>% 
    gather(key, value, -disease, -sample_type, -cohort) %>%
    mutate(key=ifelse(key=="age", "recipient age", "donor age"),
           cohort=ifelse(cohort==1, "cohort 1", "cohort 2")) %>%
    filter(disease %in% c("AML", "ALL" ,"MDS")) %>%
    group_by(sample_type, cohort) %>%
    ggplot(aes(value)) +
    geom_histogram() + 
    facet_grid(cohort~key, scales = "free_y") +
    ggtitle("Mixed Disease: Donor and Recipient Age Distribution") +
    xlab("age") +
    theme(plot.title = element_text(hjust = 0.5))
```



Proportion of Events AML/MDS with outcomes

```{r, echo=FALSE}
amlmds.outcome <- covariate.file %>%
    filter(disease %in% c("AML","MDS")) %>%
    group_by(sample_type, cohort) %>%
    summarize(OS=sum(dead_1Y),
              DRM=sum(disease_death_1Y),
              TRM=sum(TRM_1Y),
              PFS=sum(lfs_1Y),
              REL=sum(rel_1Y, na.rm=TRUE),
              GVHD=sum(GVHD_death_1Y),
              OF=sum(OF_1Y),
              INF=sum(infection_1Y)
              ) %>% 
    arrange(cohort) %>%
    ungroup() %>%
    mutate(total=c(1584, 1627, 670, 683)) %>%
    mutate_at(.vars=vars(OS, DRM, PFS, REL, TRM, GVHD, OF, INF), .funs=funs(paste0(., " (", round((./total)*100, 2), "%)"))) %>%
    select(-total) %>%
    arrange(sample_type) %>% 
    unite(sample_cohort, c("sample_type", "cohort")) %>% 
    gather(key, value, -sample_cohort) %>%
    spread(sample_cohort, value) %>% 
    rename(`Donor Cohort 1`=donor_1,
           `Donor Cohort 2`=donor_2,
           `Recipient Cohort 1`=recipient_1,
           `Recipient Cohort 2`=recipient_2) %>%
    mutate(key=factor(key, levels=c("OS", "DRM", "PFS", "REL", "TRM", "GVHD", "INF", "OF"))) %>%
    arrange(key) %>%
    as.data.frame()



rownames(amlmds.outcome) <- amlmds.outcome$key
amlmds.outcome <- amlmds.outcome %>%
    select(-key) 

amlmds.outcome %>%
    kable(format = "latex",
          caption="AML + MDS: Proportion of Events by Outcome and Cohort in Donor and Recipients", 
          booktabs=TRUE) %>%
    kable_styling(latex_options = "striped",
                  full_width=FALSE) %>%
    add_header_above(c(" ", "Donor (N/Percent %)"=2, "Recipient (N/Percent %)"=2)) %>%
    footnote(general_title="",
             general="Shown here are the proprtion of events for AML+MDS in both DISCOVeRY-BMT cohorts",
             footnote_as_chunk = TRUE,
             threeparttable = TRUE)
``` 


```{r, echo=FALSE, fig.cap="Histogram of Donor and Recipient Age Distributions in AML and MDS patients"}
covariate.file %>%
    select(disease, sample_type, cohort, age, dnrage) %>% 
    gather(key, value, -disease, -sample_type, -cohort) %>%
    mutate(key=ifelse(key=="age", "recipient age", "donor age"),
           cohort=ifelse(cohort==1, "cohort 1", "cohort 2")) %>%
    filter(disease %in% c("AML", "MDS")) %>%
    group_by(sample_type, cohort) %>%
    ggplot(aes(value)) +
    geom_histogram() + 
    facet_grid(cohort~key, scales = "free_y") +
    ggtitle("AML + MDS: Donor and Recipient Age Distribution") +
    xlab("age") +
    theme(plot.title = element_text(hjust = 0.5))
```





Proportions of Events in AMLonly with outcomes
```{r, echo=FALSE}
amlonly.outcome <- covariate.file %>%
    filter(disease %in% c("AML")) %>%
    group_by(sample_type, cohort) %>%
    summarize(OS=sum(dead_1Y),
              DRM=sum(disease_death_1Y),
              TRM=sum(TRM_1Y),
              PFS=sum(lfs_1Y),
              REL=sum(rel_1Y, na.rm=TRUE),
              GVHD=sum(GVHD_death_1Y),
              OF=sum(OF_1Y),
              INF=sum(infection_1Y)) %>% 
    arrange(cohort) %>%
    ungroup() %>%
    mutate(total=c(1247, 1282, 478, 488)) %>%
    mutate_at(.vars=vars(OS, DRM, PFS, REL, TRM, GVHD, OF, INF), .funs=funs(paste0(., " (", round((./total)*100, 2), "%)"))) %>%
    select(-total) %>%
    arrange(sample_type) %>% 
    unite(sample_cohort, c("sample_type", "cohort")) %>% 
    gather(key, value, -sample_cohort) %>%
    spread(sample_cohort, value) %>% 
    rename(`Donor Cohort 1`=donor_1,
           `Donor Cohort 2`=donor_2,
           `Recipient Cohort 1`=recipient_1,
           `Recipient Cohort 2`=recipient_2) %>%
    mutate(key=factor(key, levels=c("OS", "DRM", "PFS", "REL", "TRM", "GVHD", "INF", "OF"))) %>%
    arrange(key) %>%
    as.data.frame()

rownames(amlonly.outcome) <- amlonly.outcome$key
amlonly.outcome <- amlonly.outcome %>%
    select(-key) 

amlonly.outcome %>%
    kable(format = "latex",
          caption="AML only: Proportion of Events by Outcome and Cohort in Donor and Recipients", 
          booktabs=TRUE) %>%
    kable_styling(latex_options = "striped",
                  full_width=FALSE) %>%
    add_header_above(c(" ", "Donor (N/Percent %)"=2, "Recipient (N/Percent %)"=2)) %>%
    footnote(general_title="",
             general="Shown here are the proprtion of events for AML only in both DISCOVeRY-BMT cohorts",
             footnote_as_chunk = TRUE,
             threeparttable = TRUE)
```


```{r, echo=FALSE, fig.cap="Histogram of Recipient Age Distribution in AML and MDS patients"}
covariate.file %>%
    select(disease, sample_type, cohort, age, dnrage) %>% 
    gather(key, value, -disease, -sample_type, -cohort) %>%
    mutate(key=ifelse(key=="age", "recipient age", "donor age"),
           cohort=ifelse(cohort==1, "cohort 1", "cohort 2")) %>%
    filter(disease %in% c("AML")) %>%
    group_by(sample_type, cohort) %>%
    ggplot(aes(value)) +
    geom_histogram() + 
    facet_grid(cohort~key,  scales = "free_y") +
    ggtitle("AML Patients: Donor and Recipient Age Distribution") +
    xlab("age") +
    theme(plot.title = element_text(hjust = 0.5))
```


Proportions of Events in ALLonly with outcomes

```{r, echo=FALSE}
allonly.outcome <- covariate.file %>%
    filter(disease %in% c("ALL")) %>%
    group_by(sample_type, cohort) %>%
    summarize(OS=sum(dead_1Y),
              DRM=sum(disease_death_1Y),
              TRM=sum(TRM_1Y),
              PFS=sum(lfs_1Y),
              REL=sum(rel_1Y, na.rm=TRUE),
              GVHD=sum(GVHD_death_1Y),
              OF=sum(OF_1Y),
              INF=sum(infection_1Y)) %>% 
    arrange(cohort) %>% 
    ungroup() %>% 
    mutate(total=c(468, 483, 93, 94)) %>% 
    mutate_at(.vars=vars(OS, DRM, PFS, REL, TRM, GVHD, OF, INF), .funs=funs(paste0(., " (", round((./total)*100, 2), "%)"))) %>%
    select(-total) %>%
    arrange(sample_type) %>% 
    unite(sample_cohort, c("sample_type", "cohort")) %>% 
    gather(key, value, -sample_cohort) %>%
    spread(sample_cohort, value) %>% 
    rename(`Donor Cohort 1`=donor_1,
           `Donor Cohort 2`=donor_2,
           `Recipient Cohort 1`=recipient_1,
           `Recipient Cohort 2`=recipient_2) %>%
    mutate(key=factor(key, levels=c("OS", "DRM", "PFS", "REL", "TRM", "GVHD", "INF", "OF"))) %>%
    arrange(key) %>%
    as.data.frame()

rownames(allonly.outcome) <- allonly.outcome$key
allonly.outcome <- allonly.outcome %>%
    select(-key) 

allonly.outcome %>%
    kable(format = "latex",
          caption="ALL only: Proportion of Events by Outcome and Cohort in Donor and Recipients", 
          booktabs=TRUE) %>%
    kable_styling(latex_options = "striped",
                  full_width=FALSE) %>%
    add_header_above(c(" ", "Donor (N/Percent %)"=2, "Recipient (N/Percent %)"=2)) %>%
    footnote(general_title="",
             general="Shown here are the proprtion of events for ALL only in both DISCOVeRY-BMT cohorts",
             footnote_as_chunk = TRUE,
             threeparttable = TRUE)

```

```{r, echo=FALSE, fig.cap="Histogram of Donor and Recipient Age Distributions in ALL patients"}
covariate.file %>%
    select(disease, sample_type, cohort, age, dnrage) %>% 
    gather(key, value, -disease, -sample_type, -cohort) %>%
    mutate(key=ifelse(key=="age", "recipient age", "donor age"),
           cohort=ifelse(cohort==1, "cohort 1", "cohort 2")) %>%
    filter(disease %in% c("ALL")) %>%
    group_by(sample_type, cohort) %>%
    ggplot(aes(value)) +
    geom_histogram() + 
    facet_grid(cohort~key, scales = "free_y") +
    ggtitle("ALL Patients: Donor and Recipient Age Distribution") +
    xlab("age") +    
    theme(plot.title = element_text(hjust = 0.5))
```


Proportions of Events in MDSonly with outcomes

```{r, echo=FALSE}
mdsonly.outcome <- covariate.file %>%
    filter(disease %in% c("MDS")) %>%
    group_by(sample_type, cohort) %>%
    summarize(OS=sum(dead_1Y),
              DRM=sum(disease_death_1Y),
              TRM=sum(TRM_1Y),
              PFS=sum(lfs_1Y),
              REL=sum(rel_1Y, na.rm=TRUE),
              GVHD=sum(GVHD_death_1Y),
              OF=sum(OF_1Y),
              INF=sum(infection_1Y)) %>% 
    arrange(cohort) %>%
    ungroup() %>%
    mutate(total=c(337, 345, 192, 195)) %>%
    mutate_at(.vars=vars(OS, DRM, PFS, REL, TRM, GVHD, OF, INF), .funs=funs(paste0(., " (", round((./total)*100, 2), "%)"))) %>%
    select(-total) %>%
    arrange(sample_type) %>% 
    unite(sample_cohort, c("sample_type", "cohort")) %>% 
    gather(key, value, -sample_cohort) %>%
    spread(sample_cohort, value) %>% 
    rename(`Donor Cohort 1`=donor_1,
           `Donor Cohort 2`=donor_2,
           `Recipient Cohort 1`=recipient_1,
           `Recipient Cohort 2`=recipient_2) %>%
    mutate(key=factor(key, levels=c("OS", "DRM", "PFS", "REL", "TRM", "GVHD", "INF", "OF"))) %>%
    arrange(key) %>%
    as.data.frame()

rownames(mdsonly.outcome) <- mdsonly.outcome$key
mdsonly.outcome <- mdsonly.outcome %>%
    select(-key) 

mdsonly.outcome %>%
    kable(format = "latex",
          caption="MDS only: Proportion of Events by Outcome and Cohort in Donor and Recipients", 
          booktabs=TRUE) %>%
    kable_styling(latex_options = "striped",
                  full_width=FALSE) %>%
    add_header_above(c(" ", "Donor (N/Percent %)"=2, "Recipient (N/Percent %)"=2)) %>%
    footnote(general_title="",
             general="Shown here are the proprtion of events for MDS only in both DISCOVeRY-BMT cohorts",
             footnote_as_chunk = TRUE,
             threeparttable = TRUE)
```

```{r, echo=FALSE, fig.cap="Histogram of Donor and Recipient Age Distributions in MDS patients"}
covariate.file %>%
    select(disease, sample_type, cohort, age, dnrage) %>% 
    gather(key, value, -disease, -sample_type, -cohort) %>%
    mutate(key=ifelse(key=="age", "recipient age", "donor age"),
           cohort=ifelse(cohort==1, "cohort 1", "cohort 2")) %>%
    filter(disease %in% c("MDS")) %>%
    group_by(sample_type, cohort) %>%
    ggplot(aes(value)) +
    geom_histogram() + 
    facet_grid(cohort~key,  scales = "free_y") +
    ggtitle("MDS Patients: Donor and Recipient Age Distribution") +
    xlab("age") +
    theme(plot.title = element_text(hjust = 0.5))
```

Survival Curves for One year Outcomes




```{r, echo=FALSE, fig.width=8, fig.height=5, fig.align="left", fig.cap="Survival curves for all disease groups being tested (Mixed disease, AML+MDS, AML only, and ALL only). Cohorts 1 and 2 are shown in the left and right panel respectively. Red is disease-related mortality (DRM), green is overall survival (OS), and blue is transplant related mortality (TRM)."}
covs_surv <- covariate.file %>%
    select(sample_type,
           cohort,
           disease,
           intxsurv_1Y,
           intxrel_1Y,
           dead_1Y,
           lfs_1Y,
           rel_1Y,
           disease_death_1Y,
           TRM_1Y,
           GVHD_death_1Y,
           infection_1Y,
           OF_1Y)

covs_surv_long <- covs_surv %>%
    rename(surv_time=intxsurv_1Y,
           rel_time=intxrel_1Y,
           lfs_1y=lfs_1Y,
           rel_1y=rel_1Y) %>%
    mutate_at(.vars=vars(contains("_1Y", ignore.case = FALSE)), .funs=funs(paste0(., " ", surv_time))) %>%
    mutate_at(.vars=vars(contains("_1y", ignore.case = FALSE)), .funs=funs(paste0(., " ", rel_time))) %>%
    select(-surv_time,
           -rel_time) %>%
    gather(key, value, -sample_type, -cohort, -disease) %>% 
    separate(value, c("event", "time"), sep=" ") %>%
    mutate_at(.vars=vars(event, time), .funs=as.double) %>%
    filter(sample_type=="recipient") %>%
    mutate(key=str_replace(key, "dead_1Y", "OS"),
           key=str_replace(key, "disease_death_1Y", "DRM"),
           key=str_replace(key, "TRM_1Y", "TRM"),
           key=str_replace(key, "lfs_1y", "PFS"),
           key=str_replace(key, "rel_1y", "REL"),
           key=str_replace(key, "GVHD_death_1Y", "GVHD"),
           key=str_replace(key, "infection_1Y", "INF"),
           key=str_replace(key, "OF_1Y", "OF")) %>%
    rename(outcome=key)

library(survminer)
library(survival)

mixed.main <- covs_surv_long %>%
    filter(outcome %in% c("OS", "DRM" ,"TRM"))

m <- survfit(Surv(time=time, event=event) ~ outcome + cohort, 
        data=mixed.main) %>%
    ggsurvplot(fit=.,
               data=mixed.main,
               risk.table=FALSE,
               conf.int=FALSE,
               title="Mixed Disease Cohorts 1 and 2 Survival Curves") 

m.df <- m$data.survplot %>%
    mutate(disease="Mixed")

amlmds.main <- covs_surv_long %>%
    filter(disease %in%  c("AML", "MDS"),
           outcome %in% c("OS", "DRM" ,"TRM"))

amlonly <- covs_surv_long %>%
    filter(disease %in%  c("AML"),
           outcome %in% c("OS", "DRM" ,"TRM"))

allonly <- covs_surv_long %>%
    filter(disease %in%  c("ALL"),
           outcome %in% c("OS", "DRM" ,"TRM"))

mdsonly <- covs_surv_long %>%
    filter(disease %in%  c("MDS"),
           outcome %in% c("OS", "DRM" ,"TRM"))


m2 <- survfit(Surv(time=time, event=event) ~ outcome + cohort, 
        data=amlmds.main) %>%
    ggsurvplot(fit=.,
               data=amlmds.main,
               risk.table=FALSE,
               conf.int=FALSE,
               title="AML + MDS Cohorts 1 and 2 Survival Curves") 

m3 <- survfit(Surv(time=time, event=event) ~ outcome + cohort, 
        data=amlonly) %>%
    ggsurvplot(fit=.,
               data=amlonly,
               risk.table=FALSE,
               conf.int=FALSE,
               title="AML only Cohorts 1 and 2 Survival Curves")

m4 <- survfit(Surv(time=time, event=event) ~ outcome + cohort, 
        data=allonly) %>%
    ggsurvplot(fit=.,
               data=allonly,
               risk.table=FALSE,
               conf.int=FALSE,
               title="ALL only Cohorts 1 and 2 Survival Curves")

m5 <- survfit(Surv(time=time, event=event) ~ outcome + cohort, 
        data=mdsonly) %>%
    ggsurvplot(fit=.,
               data=mdsonly,
               risk.table=FALSE,
               conf.int=FALSE,
               title="AML + MDS Cohorts 1 and 2 Survival Curves")


m2.df <- m2$data.survplot %>%
    mutate(disease="AML + MDS")

m3.df <- m3$data.survplot %>%
    mutate(disease="AML")

m4.df <- m4$data.survplot %>%
    mutate(disease="ALL")

m5.df <- m5$data.survplot %>%
    mutate(disease="MDS")


m.new <- rbind(m.df, m2.df, m3.df, m4.df, m5.df)

m.new <- m.new %>%
    mutate(disease=factor(disease, levels=c("Mixed", "AML + MDS", "AML", "ALL", "MDS")))

p <- m.new %>%
    mutate(cohort=ifelse(cohort==1, "Cohort 1", "Cohort 2")) %>%
    ggplot(aes(time, surv, color=outcome)) +
    geom_line() + 
    facet_grid(cohort~disease) +
    ylim(c(0, 1)) +
    scale_x_continuous(name="Time", breaks=c(0,3, 6, 9, 12)) +
    labs(title="Disease, Transplant and Overall Death: All Disease Groups Cohorts 1 and 2 Survival Curves",
         y="Survival Probability") +
    geom_ribbon(aes(ymin=m.new$surv-m.new$std.err,
                    ymax=m.new$surv+m.new$std.err),
                alpha=0.2,
                linetype=0) +
    theme_bw() 
p +
    theme(legend.position = c(0.047, 0.17),
          legend.title=element_text(size=8),
          legend.text=element_text(size=6) )

```


```{r, echo=FALSE, fig.width=8, fig.height=5, fig.align="left", fig.cap="Survival curves for all disease groups being tested (Mixed disease, AML+MDS, AML only, and ALL only). Cohorts 1 and 2 are shown in the left and right panel respectively. Red is progression free survival (PFS) and teal is relapse (REL)."}
mixed.main <- covs_surv_long %>%
    filter(outcome %in% c("PFS", "REL"))

m <- survfit(Surv(time=time, event=event) ~ outcome + cohort, 
        data=mixed.main) %>%
    ggsurvplot(fit=.,
               data=mixed.main,
               risk.table=FALSE,
               conf.int=FALSE,
               title="Mixed Disease Cohorts 1 and 2 Survival Curves") 

m.df <- m$data.survplot %>%
    mutate(disease="Mixed")

amlmds.main <- covs_surv_long %>%
    filter(disease %in%  c("AML", "MDS"),
           outcome %in% c("PFS", "REL"))

amlonly <- covs_surv_long %>%
    filter(disease %in%  c("AML"),
           outcome %in% c("PFS", "REL"))

allonly <- covs_surv_long %>%
    filter(disease %in%  c("ALL"),
           outcome %in% c("PFS", "REL"))

mdsonly <- covs_surv_long %>%
    filter(disease %in%  c("MDS"),
           outcome %in% c("PFS", "REL"))


m2 <- survfit(Surv(time=time, event=event) ~ outcome + cohort, 
        data=amlmds.main) %>%
    ggsurvplot(fit=.,
               data=amlmds.main,
               risk.table=FALSE,
               conf.int=FALSE,
               title="AML + MDS Cohorts 1 and 2 Survival Curves") 

m3 <- survfit(Surv(time=time, event=event) ~ outcome + cohort, 
        data=amlonly) %>%
    ggsurvplot(fit=.,
               data=amlonly,
               risk.table=FALSE,
               conf.int=FALSE,
               title="AML only Cohorts 1 and 2 Survival Curves")

m4 <- survfit(Surv(time=time, event=event) ~ outcome + cohort, 
        data=allonly) %>%
    ggsurvplot(fit=.,
               data=allonly,
               risk.table=FALSE,
               conf.int=FALSE,
               title="ALL only Cohorts 1 and 2 Survival Curves")

m5 <- survfit(Surv(time=time, event=event) ~ outcome + cohort, 
        data=mdsonly) %>%
    ggsurvplot(fit=.,
               data=mdsonly,
               risk.table=FALSE,
               conf.int=FALSE,
               title="AML + MDS Cohorts 1 and 2 Survival Curves")


m2.df <- m2$data.survplot %>%
    mutate(disease="AML + MDS")

m3.df <- m3$data.survplot %>%
    mutate(disease="AML")

m4.df <- m4$data.survplot %>%
    mutate(disease="ALL")

m5.df <- m5$data.survplot %>%
    mutate(disease="MDS")


m.new <- rbind(m.df, m2.df, m3.df, m4.df, m5.df)

m.new <- m.new %>%
    mutate(disease=factor(disease, levels=c("Mixed", "AML + MDS", "AML", "ALL", "MDS")))

p <- m.new %>%
    mutate(cohort=ifelse(cohort==1, "Cohort 1", "Cohort 2")) %>%
    ggplot(aes(time, surv, color=outcome)) +
    geom_line() + 
    facet_grid(cohort~disease) +
    ylim(c(0, 1)) +
    scale_x_continuous(name="Time", breaks=c(0,3, 6, 9, 12)) +
    labs(title="Progression and Relapse Death: All Disease Groups Cohorts 1 and 2 Survival Curves",
         y="Survival Probability") +
    geom_ribbon(aes(ymin=m.new$surv-m.new$std.err,
                    ymax=m.new$surv+m.new$std.err),
                alpha=0.2,
                linetype=0) +
    theme_bw() 
p +
    theme(legend.position = c(0.047, 0.17),
          legend.title=element_text(size=8),
          legend.text=element_text(size=6) )

```


## Feature Selection 

Prior to genetic analyses, clinical covariates for inclusion in genome-wide survival models were selected using bidirectional stepwise regression [@Venables_2002] on Cox proportional hazard models [@cox1972] of OS, PFS, TRM and DRM using R statistical software [@r_core]. Cox proportional hazard models of OS, TRM and DRM evaluated SNPs associated with time to death with all survivors censored at 1 year post-BMT. PFS was defined as the time to disease progression or death [@therneau2013]. 

Deaths from TRM and DRM were treated as competing risks and analyzed accordingly [@fine1999]. SNP models for OS adjusted for recipient age, disease status (early/intermediate or advanced), and graft source (blood or marrow); PFS and DRM SNP models adjusted for recipient age and disease status; TRM SNP models adjusted for recipient age, graft source and body mass index (underweight/normal, overweight, or obese). Dosage data accounting for the probability of each genotype were used in all analyses of imputed data. Effect size estimates and standard errors from DISCOVeRY-BMT Cohorts 1 and 2 were compared and combined using a fixed-effects inverse variance meta-analyses in METAL [@metal]. For each SNP,  heterogeneity of effect size estimates between cohorts 1 and 2 was assessed using p-values from significance tests of heterogeneity ($P_{het}$) and $I^{2}$. Variants with $P_{het}$ < 0.05 and $I^{2}>50$ were meta-analyzed with a random effects models using meta in R [@Schwarzer_2007].

### Covariates Included In Models
1. OS -- age, disease status, graft source
2. DRM -- age, disease status
3. TRM -- age, BMI (obesity, overweight), graft source
4. PFS -- age, disease status
5. OF -- disease status, graft source
6. INF -- age, BMI, CMVpn, CMVp, CMVn
7. GVHD -- age, donor age, BMI

Disease types are included in the model, depending on the disease group the analysis was conducted on. E.g. for analyses without ALL (AML and MDS), AML dummy or MDS dummy were used. 


### Cox Proportional Hazards Model

We describe how model equations already developed to express the effects of exposure on disease rates calculated from grouped data are adapted to the continuous case. Section 5.2 introduces the 'partial likelihood' methodology for estimating regression coefficients in models in which the exposure variables are assumed to act multiplicatively on the background rates. 

### Model Validation

### Power Calculations

We conducted the first adequately powered evaluation of these candidate SNP and gene hypotheses using typed and imputed data from an existing genome-wide association study (GWAS) named Determining the Influence of Susceptibility COnveying Variants Related to one-Year mortality after BMT (DISCOVeRY-BMT) to replicate or validate these published associations.$^{71-72}$ In addition, we leveraged the available genome-wide data from DISCOVeRY-BMT and measured the aggregate association of all SNPs in the candidate genes with survival outcomes to determine how many of these candidate genes play a significant role in survival after transplant. Lastly, using publically available data, we characterized the potential functionality of each candidate SNP in relation to the gene of interest. 

We have the power to detect 

Lower bound is based off each TRM subtype being at least 0.10 (10%) of all patients.


```{r, fig.width=6, fig.height=4, fig.cap="Hazard ratios associated with survival models.", echo=FALSE}
#####power to detect main snp effects associated with survival#####################
get.delta=function(PA, propEvents, N, alpha, beta)
  exp((qnorm(1-alpha)+qnorm(beta))/(sqrt(N*propEvents*PA*(1-PA))))
propEvents=seq(0.1, 0.7, by=0.01)
delta=mapply(get.delta, 
             propEvents,
             MoreArgs=list(PA=0.40, N=2580, alpha=0.05/1000000, beta=0.20))
delta.lb=mapply(get.delta,
                propEvents,
                MoreArgs=list(PA=0.05,
                              N=2580,
                              alpha=0.05/1000000,
                              beta=0.20))
plot(propEvents,
     delta,type="l",
     axes=F,
     ylim=c(0,4.0)
     ,lwd=2,
     main=c("Power Calculations"),
     xlab="Proportion of Events",
     ylab=c("Minimum Detectable", "Hazard Ratio"))
abline(h=seq(0,4.0,by=0.25),col="gray72")
abline(v=seq(0.1,0.7,by=0.025),col="gray72")
axis(1);axis(2)
lines(propEvents,delta,lwd=2)
lines(propEvents,delta.lb,lwd=2,lty=2)
legend(0.276,
       4.0,
       lty=1:2,
       legend=c("MAF=0.40","MAF=0.05"),
       lwd=2,
       cex=1.2,
       bty="o",
       bg="white")
# df <- data.frame(propEvents, delta, delta.lb)
# df %>% filter(propEvents %in% c(propEvents[6], 0.5))
```



# Literature Review
An extensive literature search of PubMed was performed using to identify peer-reviewed scientific studies (published on or before December 30, 2016) that reported non-HLA genetic polymorphisms associated with survival outcomes after allogeneic BMT, including disease-related mortality (DRM), progression-free survival (PFS), transplant-related mortality (TRM) and/or overall survival (OS) [@Karaesmen_2017]. The PubMed search terms, filtering approach and link to all articles described herein are provided in the Supplemental Methods. 

For over a decade, researchers have conducted candidate gene association studies of patient survival outcomes after allogeneic blood or marrow transplantation (BMT). The intent of these studies was to identify genetic variants outside of the human leukocyte antigen (HLA) region that would increase knowledge about clinical management or serve as a potential target for novel therapeutics. $^{1-70}$ 

The majority of these studies tested for associations in small datasets, ranging from a few dozen to a few hundred patients and donors, included heterogeneous diseases spanning benign to malignant hematological diseases, related and/or unrelated donors with various degrees of HLA-matching and patients treated across multiple decades, from the 1980s through early 2000s. 


## Replication and Validation of Candidate Gene Studies
Results from genetic association studies should be reproduced in independent samples in order to confirm findings [@colhoun2003]. Researchers have defined two distinctive terms to describe the reproducibility based on differences between the original study population and the confirmation studies: replication and validation [@Igl_2009]. Replication is defined as the original and confirmation studies both having similar inclusion criteria (including the same ethnic/ancestral population) so that any differences between the study populations can be attributed to random variation [[@Igl_2009]. Validation is defined as the original and confirmation study populations having different inclusion criteria (including different ethnic/ancestral populations) so that any differences between the original and confirmation study could be due to systematic variation [@Igl_2009]. 

Thus, replication analyses were conducted when the original study included HLA-matched unrelated donor BMTs in patients of European ancestry. Validation analyses were performed on studies of leukemia patients of non-European ancestry, patient populations who received a BMT from a matched related donor, or patient populations that were mixed between those who received a BMT from related and unrelated donor. For studies of outcomes involving multiple hematologic malignancies, the entire DISCOVeRY-BMT study population was analyzed. If the original study population was specified as AML, ALL and/or MDS, the same disease inclusion criteria were applied so that the replication/validation study population aligned with that of the original study population. 

## Gene-Based Association Testing
VErsatile Gene-based Association Study 2 (VEGAS2) software was used for gene-based association testing [@Mishra_2015]. VEGAS2 uses $10^{6}$ Monte Carlo simulations to test the global significance of an association for sets of SNPs in defined genomic regions. VEGAS2 reports a gene-based P-value for each gene determined using individual SNP association P-values. Directional effects are not incorporated into analyses; thus, all SNPs can be aggregated without dampening an association signal.  For the gene-based replication or validation analyses, the P-values from typed and imputed SNPs in DISCOVeRY-BMT (+/- a 10kb flanking region) meta-analyses of OS, PFS, TRM and DRM were used as input into the VEGAS2 software. Gene-based P-values were calculated for donor, recipient, and R-D mismatch analyses of the full cohort (ALL, AML and MDS patients) or homogenous disease subgroups (ALL or AML or MDS patients) corresponding to the analyses performed in the original studies. 

## Functional Annotation
RegulomeDB Blood expression quantitative trait loci (eQTL) Browser, and Variant Effect Predictor (VEP)88 were used to provide functional annotation of the candidate SNPs. For each database, the raw data scores, P-values and annotations, respectively were downloaded from each website and assigned to each SNP in our list. RegulomeDB scores are categorized as follows: 1a-1f are likely to affect transcription factor binding and linked to expression of a gene target; 2a-2c are likely to affect transcription factor binding; 3a-3b are less likely to affect transcription factor binding, and > 3 has minimal binding evidence. A RegulomeDB score is assigned based on the level and evidence of functional modification attributable to the SNP in multiple cell lines from a range of tissues, with scores from 1 to 7, with 1 having the highest functional effect, supported by experimental evidence and 7 having no modifying effect. 

RegulomeDB database derives these annotations using the publically available data sets from Gene Expression Omnibus (GEO), the Encyclopedia of DNA elements (ENCODE) project and the Roadmap Epigenome Consortium. The Blood eQTL data are derived from a study of correlations between genetic variants and gene expression in over 5000 patients, with replication in almost 3000 individuals. Herein, we consider only cis-eQTLs, defined as < 250KB distance between the SNP chromosomal position and the probe midpoint for gene expression. VEP was used to determine the hypothetical functional importance of missense and nonsense variants based on SIFT, Mutation Taster and PolyPhen-2.

# Results
## DISCOVeRY-BMT Patient Characteristics
DISCOVeRY-BMT cohorts 1 and 2 include mostly 10/10 HLA-matched unrelated donors, with 281 8/8 HLA-matched donor-recipient pairs in cohort 2; all patients are of European continental ancestry. Cohorts do not differ by intensity of conditioning regimen, recipient or donor sex proportions, KPS/LPS scores. However, cohort 1 includes more ALL patients whereas cohort 2 includes more recipients with MDS. AML disease status also differs between cohorts at p<0.01 (Table 1). 

## Candidate Gene Studies of Survival Outcomes
The literature search identified 70 publications that studied a total 458 SNPs and 2 multi-allelic polymorphisms in 171 genes (Figure 1, Table S1). Studies included patients who received a transplant from an HLA-matched unrelated donor (19 articles), an HLA-matched related donor (23 articles), or both (28 articles) (Table S1). Study populations included patients and donors of European ancestry (53 articles), Asian ancestry (15 articles), or mixed genomic ancestry (2 articles) (Table S1). 

A total of 14 articles assessed genetic variation in HLA matched unrelated donor (URD) BMT patients of European ancestry, but only 7 of these articles reported significant associations (P<0.05 or an author specified significance threshold) and thus comprise our replication study (Table S2, Table S3). A total of 56 articles tested associations in either a combination of related and unrelated donors (RD-URD), only related donors (RD) and/or in non-European populations; 39 of these 56 articles reported at least one significant SNP association with survival outcome and we attempted to validate the significant findings from these 39 articles (Table S2, Table S4).


## Replication
DISCOVeRY-BMT cohorts were used to replicate published studies of European American acute leukemia or MDS patients treated with an unrelated donor BMT.1-14 Of the 7 articles whose findings we attempted to replicate, 2 articles tested multi-allelic models in NOD2/CARD155 and CCR56; 5 articles tested single SNP associations in TGFB11, CD2743, CD403, TNFSF43, HMGB14,IL1A7, IL1B7, and NOD2/CARD152 (Table 2, Figure 2, Table S3).1-7

The two NOD2/CARD15 associations were based on a three-variant R-D pair model [rs2066844 (SNP8), rs2066845 (SNP12) and rs2066847 (SNP13)] and single SNP associations with SNP13.27 The null type is when the R-D pair are homozygous common allele for all three SNPs and the effect allele combination is the presence of 1 or more minor alleles at any of the three SNPs within the R-D pair. In a study of 196 patients who received an unrelated donor BMT for AML or ALL, the NOD2/CARD15 multi-SNP model was significantly associated with OS (RR: 1.6, 95% CI 1.1-2.4, P=0.02) and TRM (RR: 1.6, 95% CI 1.1-2.4, P=0.02).5 However, in the DISCOVeRY-BMT AML and ALL patients (n=1597) treated with an unrelated donor BMT, there was no association with OS (HR: 1.03, 95% CI 0.9-1.2, P=0.72) or TRM (HR: 1.1, 95% CI 0.8-1.4, P=0.6, Figure 2, Table S3). In a study of 342 unrelated donor genotypes matched with AML or ALL patients, rs2066847 (SNP13) alone significantly increased risk of TRM and OS approximately 3-fold (P=0.001) and 2.5 (P=0.001), respectively2, however DISCOVeRY-BMT donor genotypes, did not associate with either TRM (HR: 1.17, 95% CI 0.78-1.74, P=0.45) or OS (HR: 0.98, 95% CI 0.73-1.31, P=0.89, in ALL or AML patients (Table 2, Figure 2, Table S3). 

One of the largest candidate gene studies (N=1370) showed significant associations between PFS and recipient CCR5 H1/H1 genotype (n=163), as well as with author defined genotype risk subgroups and OS.6 In DISCOVeRY-BMT, neither the CCR5 H1/H1 genotype (n=294) nor the genotype risk groups defined by H1/H16 status were significantly associated with PFS or OS (Figure 2, Table S3). The genotype risk groups tested by the authors were substantially smaller than the full cohort (Table 2). In DISCOVeRY-BMT these subgroups were approximately twice as large as those in the original study and adequately powered to detect these associations. Attempts to replicate single SNP associations in TNFSF4,3 TGFB1,1 HMGB15, IL1A7, and IL1B7 also failed (Table 2, Figure 2, Table S3). 

## Validation
We attempted to validate 36 polymorphisms in 26 genes from 39 candidate gene articles (Table S2, Table S4),15-52 including: ABCB129,32, CD1442, CTLA428,40,43-46,51, CYP2C1938, DAAM252, EP30036, ESR117, GSTA219, GZMB24, ICAM148, IL23R20,22, IL615-17, IRF337, KLRK123, LIG348, MTHFR31,35,41, MUTYH48, NOD2/CARD1525,27,30,33,50, NOS130, P2RX734, TDG48, TIRAP17, TLR442, TYMP26, and VDR18,21,39,47. These studies reported significant genetic associations with survival after transplant in patients who received a HLA-matched related donor BMT (19 articles) or had a study population including HLA-matched related and unrelated donor BMT patients, without stratification of results (17 articles). We also attempted to validate survival associations seen in non-European leukemia patients who received an unrelated donor BMT (3 articles). We present results of variants reported significant in at least two separate publications in Table 3 and Figure 3.

Our validation analyses identified only one variant associated at P<0.05. Donor variation in rs1800795 (IL-6) associated with OS (HR: 1.11, 95% CI 1.0-1.2, P=0.02) (Figure 3, Table S4). This SNP association was initially reported in a single study by Balavarca et al., 2015, (HR: 1.29, 95% CI 1.07-1.55, P=0.007) in patients with acute leukemia, CML, or lymphoma treated with a matched related or unrelated donor BMT (n=743). 

SNPs within NOD2/CARD15 were the most frequently studied and reported of all candidate gene association studies in our validation set (Table S2). NOD2/CARD15 is a susceptibility gene for inflammatory bowel disease and may be involved in Crohnâ€™s disease.27 We attempted to validate studies that reported an association of NOD2/CARD15 and survival outcomes in HLA-matched related and unrelated donor BMT patients27,30,33 or HLA-matched related donor BMT patients.25,50 Three studies reported significant findings between the presence of the NOD2/CARD15 multi-SNP polymorphism in either donor or recipient with TRM27,50  or PFS,25  however this did not validate in the DISCOVeRY-BMT cohorts (Figure 3, Table 3). There was also no significant association of the single variant rs2066842 in related/unrelated donors with PFS,30 or the single variant rs2066847 (SNP13) in recipients of related/unrelated donor BMTs with TRM (Figure 3, Table 3)33 in the DISCOVeRY-BMT cohorts.  

Due to its known functions and perceived implications in transplant biology,43 associations with multiple SNPs in CTLA4 have been tested in numerous transplant populations (Table S2), with 4 CTLA4 SNPs (rs3087243, rs231775, rs4553808, rs5742909) reported as significantly associated with survival after related or unrelated donor allogeneic BMT in acute leukemias, CML, lymphomas, MDS, and other hematological disorders (Table 3). Attempts to validate CTLA4 SNPs with DRM, PFS, OS, and TRM were unsuccessful in the DISCOVeRY-BMT cohorts (Table 3, Figure 3, Table S4). 

The remaining results of the 25 additional candidate genes containing SNPs that were tested in the DISCOVeRY-BMT cohorts are summarized in Tables S4 and 3 as well as Figure 3; no SNP associations were found at P<0.05. Importantly, the P-value distribution of the single SNP associations showed no deviation from the null expectation with 95% confidence intervals (Figure S2), suggesting we cannot reject the null hypothesis of no association with survival outcome. 

## Gene based replication and validation of previous studies
The reviewed candidate gene studies first selected genes based on their hypothesized or known function, and subsequently selected variants within that gene for single SNP or haplotype testing. Thus, while SNPs and haplotypes were tested individually for association, the hypotheses from the literature can be considered gene-based. The density of typed and imputed markers in the DISCOVeRY-BMT recipients and donors allows us to measure the aggregate effect of all SNPs within each candidate gene on survival. Genes were selected for testing from the same literature summarized above for the replication and validation SNP and haplotype analyses. VEGAS2 gene-based testing did not reveal any associations at P<0.05 with any of the survival outcomes in either the replication or validation groups (Table S5).  

## Candidate polymorphism annotation 
Candidate gene SNPs were analyzed using the RegulomeDB,86 VEP88 and Blood eQTL Browser87 databases to assess their functional characteristics and better understand their biological framework. Eighty percent of previously reported SNPs had RegulomeDB scores greater than 3 (Figure 4, Table S6), indicating that these SNPs have minimal to no effect on modifying transcription. This distribution aligns with the overall distribution of SNPs in the genome, thus the candidate SNPS are not enriched for their impact on gene expression or transcription factor binding. Our replication and validation analyses includes 2 protein coding variants, VEP shows that only, rs2066845 (SNP12) in NOD2/CARD15, is predicted to be damaging and disease causing. 

The Blood eQTL browser determines if candidate SNPs have a significant role in cis gene expression of the candidate gene. Of the 171 genes included in our literature search results, 52% have at least one significant cis-eQTL at a probe-level false discovery rate (FDR) < 0.05. On a genome-wide level, approximately 44% of genes have blood cis-eQTLs (FDR P<0.05). However, despite over half of the candidate genes having blood cis-eQTLS, only 13% of the candidate SNPs reported in these articles are blood cis-eQTLs. Thus, while blood eQTLs have been identified in these genes, they were not genotyped and analyzed in these candidate gene studies. Furthermore, almost half of the eQTLs in the candidate gene studies are correlated with expression that is not the candidate gene but rather a nearby gene. For example, rs7975232 (VDR) is an eQTL for SLC48A1 while the CTLA4 SNPs are actually eQTLs for CD28. The remaining eQTLs were correlated with expression of the candidate gene of interest, but in most cases, were also significant eQTLs for several other nearby genes (Table S6).

# Discussion
Our study aimed to replicate or validate all previous genetic association studies that investigated the non-HLA genetic effects on allogeneic BMT survival. Since previous studies selected SNPs in candidate genes, we conducted both single SNP and gene-based analyses to determine the aggregated SNP associations within candidate genes while still accounting for dependence between signals due to LD.    

The only association with P<0.05 in our replication and validation analyses using DISCOVeRY-BMT was the donor SNP rs1800795 in IL-6 with OS. As reported,18 the rationale for studying this SNP was based on the immunological function of IL-6 and two prior findings showing that it was associated with GvHD93, and response to chronic Hepatitis C virus therapy.94 We found no evidence of association at P<0.05 between donor SNP rs1800795 with death due to either GvHD or infection in the DISCOVeRY-BMT cohort (data not shown). Furthermore, rs1800795 is located in the intronic region of IL-6, has no effect on IL-6 expression or levels,95 but rather is an eQTL for two other nearby genes.95,96 

In addition to exploring this IL-6 association further we felt the validation of the CCR5 associations of H1/H1 genotype with outcome required additional efforts, as these associations were found in the largest study we attempted to validate, samples were also from CIBMTR (earlier years than our study population) and unlike many of the other studies survival effects only started to appear approximately two years post-transplant. Analyses outlined in Table 2 were performed without censor at 1 year for overall survival (median survival time 13.7 months, range <1 month-125.6 months) and progression free survival (median time 11.1 months, range <1 month-125.6 months). There were no genotype associations with either outcome at P<0.10. 

Another frequently studied gene, CTLA4, highlights the heterogeneity specific to studies of genetic variation in transplant and perhaps helps explain why we did not replicate or validate associations. rs5742909 in CTLA4 was tested for association with various survival outcomes after transplant in 6 independent studies of HLA matched-related donor-recipient pairs. In donors, the variant was found to be associated with DRM in one small study (N=120), this was the only study that tested donor genotype with DRM. Likewise, 1 out of 9 papers testing the association of rs231775 with survival outcomes measured the association of PFS with recipient rs231774 in 164 recipients (P=0.025). Despite the frequency with which these two CTLA4 variants were studied, for both SNP-outcome combinations DISCOVeRY-BMT is the only validation attempt.  These SNPs are like those of many candidate gene hypotheses, in that they have not been tested in the same genome for the same outcome in similar populations, and if they have the N is small (Table S1).

Our inability to replicate or validate previous candidate gene associations could also be due to differences in inclusion criteria with respect to disease, donor relation, or to differences in our endpoint of 1-year survival versus longer-term survival. The previous genetic associations were hypothesized to be independent of underlying hematologic disease, therefore we would expect to replicate or validate these associations in a homogeneous patient population such as DISCOVeRY-BMT. When possible we aligned our study population to the original candidate gene study (i.e. restricted to AML patients only). While DISCOVeRY-BMT focused on early 1-year survival, which may have different genetic contributions than later survival, many of the survival curves in the significant candidate gene articles show separation by genotype well before 1-year post-transplant, thus the significant published variants do not appear to correlated with only longer-term survival.

The large sample size of the DISCOVeRY-BMT provides adequate statistical power to attempt replication and validation of previously published candidate gene analyses71, however we did not reproduce these findings, similar to two other recent studies attempting to replicate previous candidate gene associations with GvHD after BMT.73,97 Other reports have also concluded that a substantial amount of the published candidate gene literature has presented false positive associations.98 

Confirming genetic association studies is vital to identify true positive genetic variants that may contribute to complex phenotypes. False associations lead to wasted time, energy and money in pursuit of confirmatory studies and could harm patients by delaying clinical discovery or by applying clinical studies too quickly without replication. Annotation of the previously reported SNP associations using publically available data show that few variants are functional; only one SNP is predicted to be damaging or deleterious, a small proportion of SNPs are correlated with gene expression, and an even smaller number are cis-eQTLs for the target gene of interest. Thus, while we did not replicate or validate these associations, the SNPs selected are not linked to functional annotation nor are they clearly related to the candidate genes. This underscores a fundamental problem with candidate gene studies which are hostage to the state of scientific knowledge at the time. Adequately powered testing of genetic associations with transplant outcomes remains critical to discovery and replication of genetic associations with the ultimate goal of improving patient outcomes.


