---
title: "chapter2"
author: "Abbas Rizvi"
date: "9/14/2018"
output:
  word_document: default
  pdf_document: default
header-includes: \usepackage{afterpage}
editor_options:
  chunk_output_type: console
---

```{r ch2_setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(kableExtra)
```

# Introduction
Genetic associations studies usually fall under two main subcategories, candidate gene association studies (CGAS) or genome wide association studies (GWAS). CGAS are association studies that investigate specific genes or regions of interest. Typically these are performed when researchers believe that the underlying biology is understood and they want to identify specific markers that contribute to genetic variation in these 'known' regions. For over a decade, researchers in the hematology and hematological transplant field have conducted CGAS that investigated the relationship between non-HLA genetics and survival outcomes after allogeneic transplant. The rationale for conducting the CGAS was to increase knowledge about clinical management or to serve as a potential target for novel therapeutics. We exhaustively searched PubMed for CGAS where the phenotype of interest was survival outcomes (DRM, TRM, OS, PFS) in patients with ALL, AML, or MDS after HLA-matched -related-donor (MRD) or -unrelated-donor (MUD) HSCT [@Karaesmen_2017]. We identified 70 studies that reported 45 SNPs in 36 genes as significantly associated with survival outcomes after transplant. DISCOVeRY-BMT was used to replicate or validate these published studies [@Karaesmen_2017]. The majority of these studies tested for associations in small datasets, ranging from a few dozen to a few hundred patients and donors, included heterogeneous diseases spanning benign to malignant hematological diseases, related and/or unrelated donors with various degrees of HLA-matching and patients treated across multiple decades, from the 1980s through early 2000s. In addition to reproducing previous findings, we were interested in agnostically evaluating whether genes that had been previously reported upon had an aggregate effect that could be detected and contributed to survival after transplant. 

# Methods

## Literature Review
An extensive literature search of PubMed was performed using to identify peer-reviewed scientific studies (published on or before December 30, 2016) that reported non-HLA genetic polymorphisms associated with survival outcomes after allogeneic BMT, including disease-related mortality (DRM), progression-free survival (PFS), transplant-related mortality (TRM) and/or overall survival (OS) [@Karaesmen_2017]. The PubMed search terms, filtering approach are described below: 

\singlespaced

```{}
(SNP[Text Word] OR ("polymorphism, genetic"[MeSH Terms] OR
("polymorphism"[All Fields] AND "genetic"[All Fields]) OR
"genetic polymorphism"[All Fields] OR
"polymorphism" [All Fields])) AND
(allo-HSCT[All Fields] OR allo-HCT[All Fields] OR
("unrelated"[All Fields] AND ("donor"[All Fields] OR 
"donors"[All Fields]) AND ("transplant"[All Fields] OR
"transplantation"[All Fields])) OR ("allogeneic"[All Fields] AND
("transplant"[All Fields] OR "transplantation"[All Fields])) OR
("hematopoietic"[All Fields] AND ("transplant"[All Fields] OR
"transplantation"[All Fields]))) AND (("mortality"[Subheading] OR
"mortality"[All Fields] OR "mortality"[MeSH Terms]) OR 
("mortality"[Subheading] OR "mortality"[All Fields] OR
"survival"[All Fields] OR "survival"[MeSH Terms]) OR
(non[All Fields] AND ("recurrence"[MeSH Terms] OR
"recurrence"[All Fields] OR "relapse"[All Fields])) OR
non-relapse[All Fields]) AND English[Language]
```

\doublespaced

\noindent The Inclusion Criteria comprised of:  
\noindent Inclusion criteria:  
1. non-HLA genes  
2. survival after BMT as phenotype  

\noindent Excluded:  
1. Non-English papers     
2. Working group studies    
3. Reviews   
4. SNPs not in build hg19    
5. Haplotypes  
6. Chronic Lymphocytic Leukemia (CML) or multiple myeloma (MM) or lymphoma only papers  
7. Autosomal only  
8. Microsatellites, CNVs, VNTRs, or other variation markers 

## Defintions of Replication and Validation
In principle, results from CGAS or GWAS should be reproduced in an independent study to confirm findings [@colhoun2003; @martin_2016]. Two distinctive terms have gained popularity amongst researchers to describe reproducibility -- specifying if there are differences between the original population that was studied and the confirmation study: replication and validation [@Igl_2009]. Replication is deemed to be when the inclusion criteria are near or completely identical (i.e. same ancestral population), so that any differences between the samples in the study can attributed to random variation [@Igl_2009]. Validation reproducibility is when the original and confirmation study have similar but slightly different inclusion criteria (i.e. different ancestral populations. In the validation case, the underlying differences between the original and confirmation study can be attributed to systematic variation [@Igl_2009].

Thus, replication analyses were conducted when the original study included HLA MUD-HSCT in patients of European ancestry. Validation analyses were performed on studies of leukemia patients of non-European ancestry, patient populations who received MRD-HSCT, or patient populations that were mixed between those who received a MRD-HSCT and MUD-HSCT [@Karaesmen_2017]. For studies of outcomes involving multiple hematologic malignancies, the entire DISCOVeRY-BMT study population was analyzed. If the original study population was specified as AML, ALL and/or MDS, the same disease inclusion criteria were applied so that the replication/validation study population aligned with that of the original study population. 

![Pipeline performed for reproducing previous candidate gene association study literature](~/Desktop/figures/repval_fig1.png)

## Genotyping data
All samples were genotyped using the Illumina Human OmniExpress BeadChip and the Illumina HumanExome BeadChip (University of Southern California Genomics Facility). In total, 637,655 and 632,823 SNPs from the OmniExpress BeadChip were available for imputation in cohorts 1 and 2, respectively, using 1000 Genomes Project phase 3. The missing genotypes were imputed using IMPUTE2 software [@Howie_2009]. QCTOOL was used to remove imputed genotypes with an info score > 0.7, certainty > 0.7, and a minor allele frequency $\geq$ 0.005. To test the joint effect of recipient and donor genetic variation, the recipient-donor (R-D) *mismatch* genome computed by taking the absolute value of the difference of minor alleles between recipient and donor at each SNP. For example, at a given SNP where the recipient is homozygous minor (2 minor alleles) and the donor is heterozygous (1 minor allele), the mismatch allele dosage would be $|2-1|=1$. Rs2066847 (SNP13) in *NOD2/CARD15* was the only variant analyzed from the Illumina HumanExome, as it was not typed on the OmniExpress chip or available following imputation.

## Genetic Models
Prior to genetic analyses, clinical covariates for inclusion in genome-wide survival models were selected using bidirectional stepwise regression [@Venables_2002] on Cox proportional hazard models [@cox1972] of OS, PFS, TRM and DRM using R statistical software [@r_core]. Cox proportional hazard models of OS, TRM and DRM evaluated SNPs associated with time to death with all survivors censored at 1 year post-HSCT  [@therneau2013]. PFS was defined as the time to disease progression or death. The covariates that were included in the model were (see appendix for R code):   
1. OS -- recipient age, disease status (early or advanced), graft source (bone marrow or peripheral blood)  
2. DRM -- recipient age, disease status (early or advanced)   
3. TRM -- recipient age, body mass index (BMI) (underweight, normal, overweight, obese), graft source (bone marrow or peripheral blood)    
4. PFS -- recipient age, disease status (early or advanced)  

Depending on the disease group, disease types were adjusted for by creating an indicator variables. E.g. For analyses without ALL (AML and MDS), the indicator variable AML dummy and MDS dummy were added to the model. Competing risk models [@fine1999] were used for TRM and DRM (because a patient can only die once). Dosage data accounting for the probability of each genotype were used in all analyses of imputed data. Effect size estimates and standard errors from DISCOVeRY-BMT Cohorts 1 and 2 were compared and combined using a fixed-effects inverse variance meta-analyses in METAL [@metal]. For each SNP, heterogeneity of effect size estimates between cohorts 1 and 2 was assessed using p-values from significance tests of heterogeneity ($P_{het}$) and $I^{2}$. Variants with $P_{het}$ < 0.05 and $I^{2}>50$ were meta-analyzed with a random effects models using meta [@Schwarzer_2007] in R.

### Multi-allelic models
*NOD2/CARD15* was the most common gene studied in the HSCT literature. The associations were based on a 3-variant R-D pair model (rs2066844 [SNP8], rs20666845 [SNP12], rs2066847 [SNP13]) and single SNP associations with SNP13 [@holler_2004]. The null type was when the R-D pair are homoyzgous common allele for all 3 SNPs and the effect allele combination was considered the presence of $/geq$ 1 minor allele in any of the 3 SNPs within the R-D pair. *CCR5* haplotype was studied in one replication study [@mcdermott2010]. The risk H1/H1 haplotype was defined as presence of homozgous genotype for the major allele at rs1799987 (AA), rs1800023 (AA), rs333 (ACAGTCAGTATCAATTCTGGAAGAATTTCCAG); individuals not homozygous common were considered null. rs333, a 32-basepair deletion, was not typed or imputed in the DISCOVeRY-BMT cohort, however we selected a proxy SNP (rs1133418) in strong linkage disequilibrium ($r^2=0.97$) with rs333. The presence of the G allele in the proxy SNP corresponds to ACAGTCAGTATCAATTCTGGAAGAATTTCCAG in rs333. McDermott and colleagues (2010) defined 3 risk subgroups:  
1. R-D Group 1 -- R-D pairs lacked *CCR5* H1/H1 haplotype    
2. R-D Group 2 -- donors only had *CCR* H1/H1 haplotype  
3. R-D Group 3 -- recipients only had *CCR5* H1/H1 haplotype  

\noindent The imputed data for *CCR5* was in the IMPUTE2 output data chunk for chromosome 3 (`BMT093013_forImpute.chr3-45000000-50000000.impute2`). QCTOOL was used to extract the data out from the IMPUTE2 and output it into the a VCF file. The location for these files was stored on the University at Buffalo Computational Center Resource (UB CCR) supercomputer. See Appendix for full code on analysis. 

\singlespacing
```{bash, eval=FALSE}
qctool \ 
    -g ./BMT093013_forImpute.chr3-45000000-50000000.impute2 \
    -s ./BMT093013_forImpute.chr16-50000000-55000000.impute2_samples \
    -og ./ccr5_rep_dosages_threshold.vcf \
    -incl-rsids ccr5_snps.txt 
```

\doublespacing
\noindent The data wrangling to prepare the data for these models was done in R. The Cox survival models were written and automated with custom R code that leveraged the survival package [@therneau2013] (see Appendix for code used for the full analysis). 


## Gene-based assocation testing
VErsatile Gene-based Association Study 2 (VEGAS2) software was used for gene-based association testing [@Mishra_2015]. VEGAS2 uses $10^{6}$ Monte Carlo simulations to test the global significance of an association for sets of SNPs in defined genomic regions. VEGAS2 reports a gene-based P-value for each gene determined using individual SNP association P-values. Directional effects are not incorporated into analyses; thus, all SNPs can be aggregated without dampening an association signal. For the gene-based replication or validation analyses, the P-values from typed and imputed SNPs in DISCOVeRY-BMT ($\pm$ a 10kb flanking region) meta-analyses of OS, PFS, TRM and DRM were used as input into the VEGAS2 software. Gene-based P-values were calculated for donor, recipient, and R-D mismatch analyses of the full cohort (ALL, AML and MDS patients) or homogeneous disease subgroups (ALL or AML or MDS patients) corresponding to the analyses performed in the original studies [@Karaesmen_2017]. 

To run VEGAS2, a flat text file is needed that has two unlabeled columns (rsid and GWAS P-value [$P_{meta}$]). 

\noindent For example:  

\singlespaced
```{bash, eval=FALSE}
#snps.txt
rs6696752   0.827182998293298
rs72638700  0.874653327370856
```

\doublespaced

\noindent And then a simple command prompting VEGAS2 is run on the command line:  

\singlespaced
```{bash, eval=FALSE}
vegas2 \
    snps.txt \
    -pop 1000GEURO \
    -subpop EURO \
    -genesize 10kbloc \
    -top 100 \
    -sex BothMnF \
    -max 1000000 \
    -out ./results/output_V2out 
```

\doublespaced
\noindent VEGAS2 analyses were using SNPs from all of the identified genes and p-values from DISCOVeRY-BMT (for all outcomes and 3 genomes) on UB CCR (see Appendix). 

## Functional Annotation
To gain a deeper understanding of genetic associations, often bioinformatics approaches leverage the plethora of curated, publically available datasets and databases [@gallagher_2018]. The databases used in this study were RegulomeDB [@Boyle_2012], Blood expression quantitative trait loci (eQTL) Browser [@westra_2013], and Variant Effect Predictor (VEP) [@vep_2010], which were used to annotate the SNPs in the candidate genes. For RegulomeDB and Blood eQTL Browser, the full database consisting of raw data scores, P-values, and annotations were downloaded. VEP annotation was done directly on VEP's website. 

RegulomeDB stores variant annotation with known and predicted regulatory DNA elements in humans. The elements include DNase hypersensitivity, transcription factor (TF) binding sites, and promoter regions that have been characterized to alter transcription [@Boyle_2012]. RegulomeDB database compiled these annotations using the publically available data sets from the Encyclopedia of DNA elements (ENCODE) project and the Roadmap Epigenome Consortium and Gene Expression Omnibus (GEO). RegulomeDB scores are the following: 1a-1f likely to affect TF binding and linked to altering expression of target genes; 2a-2c likely to affect TF binding; 3a-3b less likely to affect TF binding, and > 3 has minimal binding evidence. RegulomeDB DB scores are assigned (only one per SNP) based on the level of effect and evidence of functional modification that is attributable to the SNP across multiple cell lines from differing tissues. The scores are supported by experimental evidence -- with scores from 1 to 7, with 1 having the greatest functional effect, and scores of 7 show no evidence of having modifying effects.

Blood eQTL database was built from a study (N=5000) that investigated the correlation between genetic expression and genetic polymorphisms and replicated the results in an independent study (N $\approx$ 3000), @westra_2013]. We only considered *cis*-eQTLs, defined as < 250KB distance between the SNP chromosomal position and the probe midpoint for gene expression. Furthermore, VEP was used to determine the hypothetical (predicted) functional importance of missense and nonsense variants based on SIFT, Mutation Taster and PolyPhen-2 software.

# Results

## Candidate Gene Studies of Survival Outcomes
The literature search identified 70 publications that studied a total 458 SNPs and 2 multi-allelic polymorphisms in 171 genes (**Figure 2.1**). Studies included patients who received a transplant from an MUD-HSCT (19 articles), a MRD-HSCT (23 articles), or both (28 articles) (**Table 2.1**). Study populations included patients and donors of European ancestry (53 articles), Asian ancestry (15 articles), or mixed genomic ancestry (2 articles) (**Table 2.1**).  

\afterpage{
\begin{landscape}

```{r, echo=FALSE}
x <- readxl::read_xlsx("~/Google Drive/OSU_PHD/dissertation/code/chapter2/repval_supps.xlsx",sheet=2)

colnames(x) <- c("SNP",
                 "Gene",
                 "Reports of SNPs with any HSCT outcome",
                 "Significant SNPs with any HSCT outcome",
                 "Reports of SNPs European ancestry MUD-HSCT populations",
                 "Significant SNPs in European ancestry MUD-HSCT populations",
                 "Reports of SNPs tested non-European ancestry and/or MRD-HSCT populations",
                 "Significant SNPs in non-European ancestry and/or MRD-HSCT populations")

x %>%
    filter(`Reports of SNPs with any HSCT outcome`>=2 | `Reports of SNPs European ancestry MUD-HSCT populations`>=2 | `Reports of SNPs tested non-European ancestry and/or MRD-HSCT populations`>=2 | Gene=="CCR5") %>%
    mutate(SNP=ifelse(SNP == "grp3vsgrp1", "^ R-D Group 3 vs R-D Group 1", SNP),
           SNP=ifelse(SNP=="grp3vsgrp2", "^ R-D Group 3 vs R-D Group 2", SNP),
           Gene=ifelse(Gene=="NOD2/CARD15","NOD2/ CARD15", Gene)) %>%
    kable(format="latex", 
          caption = "Count of reports with SNPs that were studied at least twice (in addition to CCR5 studies) that were attempted for replication or validation in DISCOVeRY-BMT.") %>%
    column_spec(width="8.15em", column=1) %>%
    column_spec(width="6em", column=2) %>%
    column_spec(width="9.2em", column=3) %>%
    column_spec(width="9.2em", column=4) %>%
    column_spec(width="9.4em", column=5) %>%
    column_spec(width="9.4em", column=6) %>%
    column_spec(width="9.1em", column=7) %>%
    column_spec(width="9.1em", column=8) %>%
    kable_styling(font_size=7,
                  latex_options = "striped") %>%
    footnote(general_title="",
             general=c("^ CCR5 H1/H1 genotypes and risk groups defined using multiallelic models described in the original publication (McDermott et al 2010) and in the main text.", "3 SNPs* -- The NOD2/CARD15 3 SNP multiallelic model. Described in the text.")) 
```

\end{landscape}
}

\normalsize

\doublespacing

A total of 14 articles assessed genetic variation in HLA MUD-HSCT patients of European ancestry, but only 7 of these articles reported significant associations ($P < 0.05$ or an author specified significance threshold) and thus comprise our replication study (**Table 2.1**, **Figure 2.2**). A total of 56 articles tested associations in either a combination of MRD and MUD-HSCT, only MRD and/or in non-European populations; 39 of these 56 articles reported at least one significant SNP association with survival outcome and we attempted to validate the significant findings from these 39 articles (**Table 2.1**, **Figure 2.3**) [@Karaesmen_2017].

## Replication
To reproduce previous reported results, DISCOVeRY-BMT was used to replicate findings that comprised of acute leukemia or MDS patients with European ancestry, treated with MUD-HSCT [@Karaesmen_2017]. Seven articles were included in the replication analyses. Multi-allelic models, *CCR5* and *NOD2/CARD15*, were tested in 2 articles; single SNP associations in  *CD274*, *CD40*, *HMGB1*, *IL1A*, *IL1B*, *NOD2/CARD15*, *TGFB1*, and *TNFSF4* were tested in 5 articles (**Table 2.1**, **Figure 2.2**) [@Karaesmen_2017].  

\begin{figure}
    \centering
    \includegraphics[width=\linewidth, height=6in]{~/Desktop/figures/Replication_Fig.png}
    \caption[Replication attempts of previously reported significant candidate gene-association studies in DISCOVeRY-BMT.]{Replication attempts of previously reported significant candidate gene-association studies in DISCOVeRY-BMT. Survival association P values as reported in previous literature (Right panel) and replication attempts of these associations in DISCOVeRY-BMT cohort (Left panel) are shown as data points. Vertical panels indicate the genes that these polymorphisms and haplotypes are located in or close to as reported by the previous studies. Shapes represent associations with survival outcomes OS, PFS, or TRM; colors correspond to donor, recipient, or donor-recipient mismatch polymorphisms. The size of the point represents the sample size of the study, with larger points reflecting a bigger sample size. Shown on y-axis are the 9 polymorphisms from the literature reporting associations at $P<.05$ with OS, PFS, or TRM by 1 or more previously published studies; the x-axis is the $\log_{10}$(P value). The red vertical lines in (left panel) and (right panel) indicate $P<.05$. Details on the haplotypes are described in “Results” under “Replication.”}
    \label{fig:rep_fig}
\end{figure}

As previously discussed, candidate genes and candidate SNPs are selected due to *a priori* biological knowledge of the phenotype. The general thought in HSCT is that patients are either dying from their disease or from the transplant. And transplant related causes are mostly infection, organ failure, or GVHD. Relevant biological pathways that are included in these phenotypes are likely to involve immunological and inflammatory pathways. Indeed, the most frequently studied gene was *NOD2/CARD15*, which is a susceptibility gene for inflammatory bowel disease and may be involved in Crohn’s disease [@holler_2008] (**Table 2.1**). The two *NOD2/CARD15* associations were based on a three-variant R-D pair model [rs2066844 (SNP8), rs2066845 (SNP12) and rs2066847 (SNP13)] and single SNP associations with SNP13 [@holler_2008]. The null genotype is when the R-D pair are homozygous common allele for all 3 SNPs and the effect allele combination is the presence of 1 or more minor alleles at any of the 3 SNPs within the R-D pair. The *NOD2/CARD15* multi-SNP model was significantly associated with OS (RR: 1.6, 95% CI 1.1-2.4, $P=0.02$) and TRM (RR: 1.6, 95% CI 1.1-2.4, $P=0.02$) in a study of 196 patients who received a MUD-HSCT for AML or ALL. However, no associations were found for OS (HR: 1.03, 95% CI 0.9-1.2, $P=0.72$) or TRM (HR: 1.1, 95% CI 0.8-1.4, $P=0.6$) in DISCOVeRY-BMT, which is a larger study (N=1597) with AML and ALL patients treated with MUD-HSCT (**Figure 2.2**). Furthermore, a study of 342 AML or ALL patients after MUD-HSCT (@holler_2008) reported donor genotype rs2066847 (SNP13) significantly increased risk of TRM and OS approximately 3-fold ($P=0.001$) and 2.5 ($P=0.001$), respectively. When this SNP was tested in DISCOVeRY-BMT donors, no associations were found with either TRM (HR: 1.17, 95\% CI 0.78-1.74, $P=0.45$) or OS (HR: 0.98, 95% CI 0.73-1.31, $P=0.89$, in ALL or AML patients) (**Figure 2.2**). 

One of the largest CGAS (N=1370) reported significant associations for *CCR5* H1/H1 genotype (N=163) in recipients [@mcdermott2010]. McDermott and colleagues also defined genotype risk subgroups and OS (**Figure 2.2**, **Table 2.1**). These associations were tested in DISCOVeRY-BMT and neither *CCR5* H1/H1 genotype (N=294) nor the genotype risk groups defined by H1/H1 status were significantly associated with PFS or OS (**Figure 2.2**, **Table 2.1**). The genotype risk groups (Group 3 vs Group 1 and Group 3 vs Group 2) were substantially smaller than the full cohort. We tested these in DISCOVeRY-BMT and found no associations. Considering the fact that DISCOVeRY-BMT cohorts were approximately twice as large as those in the original study and adequately powered to detect these associations (**Figure 1.11**), these risk group associations were not real [@Karaesmen_2017]. In DISCOVeRY-BMT these subgroups were approximately twice as large as those in the original study and adequately powered to detect these associations. 

DISCOVeRY-BMT was unable to replicate another large CGAS of 1170 patients [@jindra_2016], which reported an association between rs10912564 (*TNFSF4*) and TRM ($P=0.017$), OS ($P=0.022$), and PFS (HR: 0.8, 95% CI [0.9-1.2], $P=0.03$) (**Figure 2.2**). Similarly DISCOVeRY-BMT could not replicate rs2249825 in *HMGB1* [@kornblit_2010, N=276], rs1800469 in *TGFB1* [@bolanos_2016, N=493], rs755622 in *MIF* [@chang_2009, N=454], nor SNPs in *IL1A* and *IL1B* (**Figure 2.2**). 

## Validation

Validation attempts were conducted on 36 genetic polymorphisms in 26 genes from 39 previously published CGAS (**Table 2.1**). The genes that were included: *ABCB1*, *CD14*, *CTLA4*, *CYP2C19*, *DAAM2*, *EP300*, *ESR1*, *GSTA2*, *GZMB*, *ICAM1*, *IL23R*, *IL6*, *IRF3*, *KLRK1*, *LIG3*, *MTHFR*, *MUTYH*, *NOD2/CARD15*, *NOS1*, *P2RX7*, *TDG*, *TIRAP*, *TLR4*, *TYMP*, and *VDR* (**Figure 2.3**) [@Karaesmen_2017]. Each of these studies reported at least one significant genetic associations with survival in patients who received a HLA MRD-HSCT (19 articles) or had a study population including MRD- and MUD-HSCT patients, without stratification of results (17 articles). Validation attempts for survival associations reported in non-European leukemia patients who received an MUD-HSCT (3 articles) (**Table 2.1**). The variants that had been reported to be associated to survival outcomes after transplant and our attempts to validate the results in DISCOVeRY-BMT are shown in **Figure 2.3**.

\begin{figure}
    \centering
    \includegraphics[width=\linewidth, height=6in]{~/Desktop/figures/Validation_Fig.png}
    \caption[Validation attempts of previously reported significant candidate gene-association studies.]{Validation attempts of previously reported significant candidate gene-association studies. Survival association P-values as reported in previous literature (Right panel) and validation attempts of these associations in DISCOVeRY-BMT cohort (Left panel) are shown as data points. Vertical panels indicate the genes that these 17 polymorphisms are located in or closest to as reported by the previous studies. Shapes represent associations with survival outcomes DRM, OS, PFS, and TRM; colors correspond to donor, recipient, or donor-recipient mismatch polymorphisms. The size of the point represents the sample size of the study, with larger points reflecting a bigger sample size. Shown on y-axis are the 17 polymorphisms from the literature reporting associations at $P < .05$ with OS, PFS, or TRM by 1 or more previously published studies; the x-axis is the $\log_{10}$(P-value). The red vertical lines in (left panel) and (right panel) indicate $P < .05$. Details on the haplotypes are described in “Results” under “Validation."}
    \label{fig:val_fig}
\end{figure}

Only one of our validation attempts were successfully reproduced -- where donor genotype rs1800795 (*IL-6*) was associated with OS (HR: 1.11, 95% CI 1.0-1.2, $P=0.02$) (**Figure 2.3**, note: this did not reach genome-wide significance at the $P < 5x10^{-08}$ threshold). The rs1800795 association to OS (HR: 1.29, 95% CI 1.07-1.55, $P=0.007$) was reported in a single study in patients with acute leukemia, CML, or lymphoma (N=743) treated with MRD- or MUD-HSCT [@balavarca_2015]. 

Similar to our replication set, *NOD2/CARD15* was the most frequently studied gene and reported the most of all CGAS in our validation set (**Table 2.1**). Three studies reported associations between the presence of *NOD2/CARD15* multi-SNP polymorphism and either TRM [@holler_2004] or PFS. None of these studies were validated in DISCOVeRY-BMT cohorts (**Figure 2.3**). Likewise, DISCOVeRY-BMT was unable to validate single SNP analyses in *NOD2/CARD15* for rs2066842 in MRD- or MUD-HSCT donors with PFS or for TRM at rs2066847 (SNP13) in recipients of MRD- or MUD-HSCT (**Figure 2.3**).  


*CTLA4* is a member of the immunoglobulin superfamily that is expressed by activated T-cells and transmits inhibitory signal to T=cells. Due to these known functions and perceived implications in transplant biology, associations with multiple SNPs in *CTLA4* have been tested in numerous transplant populations (**Table 2.1**) [@perez_garcia_2007]. In previous CGAS, four *CTLA4* SNPs (rs3087243, rs231775, rs4553808, rs5742909) were significantly associated with survival after MRD- or MUD-HSCT in patients with acute leukemia, CML, lymphomas, MDS, and other hematological disorders (**Table 2.1**). Attempts to validate CTLA4 SNPs with DRM, PFS, OS, and TRM were unsuccessful in the DISCOVeRY-BMT cohorts (**Figure 2.3**). The remaining results of the 25 additional candidate genes containing SNPs that were tested in the DISCOVeRY-BMT cohorts are summarized in **Figure 2.3**; no SNP associations were found at $P<0.05$. Importantly, the P-value distribution of the single SNP associations showed no deviation from the null expectation with 95% confidence intervals (**Figure 2.4**), suggesting we cannot reject the null hypothesis of no association with survival outcome [@Karaesmen_2017]. 

\begin{figure}
    \centering
    \includegraphics{~/Desktop/figures/ch2_cg_qqplot.png}
    \caption[Quantile-quantile (QQ) plot of SNP p-values in DISCOVeRY-BMT for all previously studied SNPs.]{Quantile-quantile (QQ) plot of SNP p-values in DISCOVeRY-BMT for all previously studied SNPs. From left to right, vertical panels represent the donor, recipient, and mismatch genome, horizontal panel represents the survival outcome that was tested, from top to bottom overall survival (OS), progression free survival (PFS) and transplant related mortality (TRM). Dashed red lines indicate 95\% confidence intervals}
    \label{fig:qq_genes}
\end{figure}

## Gene based replication and validation of previous studies
From the previous literature, candidates genes were first selected genes based on their hypothesized or known function, and subsequently authors selected variants within that gene for single SNP or haplotype testing [@Karaesmen_2017]. Thus, while SNPs and haplotypes were tested individually for association, the hypotheses from the literature can be considered gene-based. The density of typed and imputed markers in the DISCOVeRY-BMT recipients and donors allows us to measure the aggregate effect of all SNPs within each candidate gene on survival. Genes were selected for testing from the same literature summarized above for the replication and validation SNP and haplotype analyses. VEGAS2 gene-based testing did not reveal any associations at $P < 0.05$ with any of the survival outcomes in either the replication or validation groups [@Karaesmen_2017].

## Candidate polymorphism annotation 
Candidate gene SNPs were analyzed using the RegulomeDB, VEP and Blood eQTL Browser databases to assess their functional characteristics and better understand their biological framework. Eighty percent of previously reported SNPs had RegulomeDB scores greater than 3 (**Figure 2.5**), indicating that these SNPs have minimal to no effect on modifying transcription. This distribution aligns with the overall distribution of SNPs in the genome, thus the candidate SNPS are not enriched for their impact on gene expression or transcription factor binding. Our replication and validation analyses includes 2 protein coding variants, VEP shows that only, rs2066845 (SNP12) in *NOD2/CARD15*, is predicted to be damaging and disease causing. 

We were interested in better understanding the biological framework of candidate SNPs that had previously been identified. The candidate SNPs were analyzed using RegulomeDB, Blood eQTL Browser, and VEP to assess functional characteristics. The majority ($geq$ 80%) candidate SNPs had RegulomeDB scores greater than 3 (**Figure 2.5**). This indicates that the SNPs had minimal to no evidence of modifying DNA transcription. The observed distribution coincides with the overall distribution of SNPs in the genome. Therefore, from our point of view, the candidate SNPs are not enriched in terms of impact on TF binding or gene expression. VEP analyses revealed that the candidate SNPs in our replication and validation analyses only included 2 protein coding variants, and that rs2066845 (SNP12) in *NOD2/CARD15* is predicted to be damaging and disease causing [@Karaesmen_2017].

```{r, echo=FALSE, fig.scap="RegulomeDB score distribution of previously studied polymorphisms.", fig.cap="RegulomeDB score distribution of previously studied polymorphisms. RegulomeDB categories are shown on the x-axis; counts of SNPs falling into RegulomeDB score category are shown on the y-axis. Blue portion of the bar indicates the counts of SNPs that were tested but not reported significant; red portion shows the counts of SNPs that were reported significant at least once. Score descriptions are given below the image. 1b indicates eQTL 1 transcription factor (TF) binding 1 any motif 1 DNase footprint 1 DNase peak; 1d, eQTL 1 TF binding 1 any motif 1 DNase peak; 1f, eQTL 1 TF binding/DNase peak; 2a, TF binding 1 matched TF motif 1 matched DNase footprint 1 DNase peak; 2b, TF binding 1 any motif 1 DNase footprint 1 DNase peak; 2c, TF binding 1 matched TF motif 1 DNase peak; 3a, TF binding 1 any motif 1 DNase peak; 4, TF binding 1 DNase peak; 5, TF binding or DNase peak; 6, motif hit; 7, no evidence."}
cg.regDB <- read.table("~/Google Drive/OSU_PHD/dissertation/code/chapter2/cg_regulomedb.txt", header = TRUE, sep="\t")

p <- cg.regDB %>%
    na.omit() %>%
    ggplot(aes(x=fct_rev(RegulomeDB_Score), y=Count, fill=pval_cat) ) +
    geom_col() +
    scale_fill_manual(values=c("black", "gray70")) +
    guides(fill=guide_legend(title=NULL), cex=1) + 
    xlab("RegulomeDB Score") +
    ylab("Number of previously reported SNPs") + 
    labs(fill="Reported Association to Survival") +
    coord_flip() + 
    theme_bw() +
    theme(legend.position = c(0.7, 0.9), 
          legend.text=element_text(size=14),
          axis.text =element_text(size=14),
          axis.title=element_text(size=16))

p
```

We looked at the Blood eQTL browser to determine if the candidate SNPs played a roll in *cis* gene expression of the candidate genes. Rough half ($\approx 52\%$) of the 171 genes that are included from the literature search had at least one significant (probe level false discovery rate, FDR $< 0.05$) *cis*-eQTL. And when compared to the entire genome, approximately 44% of genes have blood *cis* eQTLs (FDR $P<0.05$). However, when we look at the the candidate SNPs, only 13% are blood *cis*-eQTLs. So while blood eQTLs have indeed been identified in these genes, the SNPs that previous literature studied, were not the ones that affected expression. Furthermore, almost half of the eQTLs in the CGAS are correlated with expression, they alter expression in neighboring or nearby genes rather than the gene they are located in. For example, rs7975232 (*VDR*) is an eQTL for *SLC48A1* while the *CTLA4* SNPs studied are eQTLs for *CD28* [@Karaesmen_2017]. The remaining eQTLs were correlated with expression of the candidate gene of interest, but in most cases, were also significant eQTLs for several other nearby genes [@Karaesmen_2017].

# Discussion
In this study, we aimed to replicate or validate all existing CGAS literature that investigated the relationship between non-HLA genetic effects on survival after allogeneic HSCT [@Karaesmen_2017]. Considering that the main purpose of CGAS is to select SNPs due to known function at the gene level, we conducted single SNP analyses and gene-based analyses to determine the joint effects of marginally associated SNPs within candidate genes while considering the underlying genetic correlation structure due to LD. 

Donor rs1800795 in *IL6* was the only association (OS) that was able to be reproduced using DISCOVeRY-BMT (only in the validation set; no candidate SNPs replicated). The original authors [@balavarca_2015] studied *IL6* due to its known immunological function, as well as the fact that two prior studies reported significant associations in different phenotypes: to GVHD [@cavet_2001] and chronic hepatitis C virus therapy [@yee_2009]. No evidence of an association of rs1800795 at $P < 0.05$ in DISCOVeRY-BMT with death due to GVHD and/or infection (data not shown). Furthermore, rs1800795 is found in an intronic region of *IL6*, and has no effect on *IL6* expression or levels, but rather is an eQTL for two other nearby genes [@snipa, @gtex]. We made an extra effort to validate interesting CGAS findings, such as in *CCR5* and the H1/H1 risk groups from McDermott et al. 2010. These associations were found in the largest study we attempted to validate, as well as having samples from CIBMTR (earlier years than DISCOVeRY-BMT), and the reported effects started two years after HSCT. We were not able to validate these associations with either outcome at $P < 0.10$ (the threshold McDermott 2010 used, **Figure 2.3**).

*CTLA4* was another gene that was studied frequently (**Table 2.1**), this gene is a good example of the heterogeneity to studies of genetic polymorphisms in HSCT and may be a plausible explanation of why were unable to reproduce any of the previous associations. The SNP rs5742909 in *CTLA4* was studied in 6 independent studies of HLA MRD-HSCT donor-recipient pairs. For donors, rs5742909 was reported to be associated with DRM in a smaller study (N=120). The lack of consistency between studies is apparent -- previously two *CTLA4* SNPS, rs231775 and rs231774 were reported associated in recipients for PFS ($P=0.025$) -- and before this study, only 1 of 9 studies tried to validate these results. These SNPs are like many candidate gene hypotheses such that they have not been tested in the same genome for the same outcome in similar populations, and if they have, the sample size is very small. 

The inclusion criteria with respect to disease, donor relation, or to differences in our endpoint of 1-year survival versus longer-term survival could be attributed as to why previous results are not reproducible. As previous genetic associations were hypothesized to be independent of the underlying disease, one may expect that results should be reproducible in a homogeneous patient populations such as DISCOVeRY-BMT, which unfortunately was not the case. This study meticulously attempted to align the study population in DISCOVeRY-BMT to the original studies (i.e. subset the full population for this only with AML). DISCOVeRY-BMT primarily focuses on 1-year survival after HSCT and this may have different genetic effects than later survival, however, the survival curves from the CGAS often showed separation of death occurring well into the first year.  DISCOVeRY-BMT has a large sample size that is adequately powered [@lsc_2015] to reproduce results from previously published CGAS. However, we were unable to reproduce almost all of the findings. This is in agreement with other studies in this field, Particularly in the context of death due to GVHD as an outcome [@chien_2012; @martin_2016]. Several other reports have also concluded that published CGAS have presented false positive associations [@Ioannidis_2001]. 

Confirmatory studies are vital to identify true positive genetic variants that are associated to complex phenotypes. False positives may lead researchers to allocate resources and time to research that will lead to a dead end. In the long-term this may delay treatment to patients or discovery of markers that would be relevant in improving survival after HSCT. As only one SNP had a predicted damage or deleterious effect and that a only a small proportion of the previous 70 studies conducted research on SNPs that correlated with gene expression, we conclude that the transplant field can look agnostically at the entire genome to hone in on potential candidate genes and SNPs [@Karaesmen_2017]. And while we could not reproduce previous associations, the SNPs selected are not linked to functional annotation nor are they clearly related to the candidate genes. This underscores a fundamental flaw with CGAS which are contingent on the scientific knowledge at the time. We assert that by using adequately powered testing (i.e. larger studies, GWAS) with transplant will be vital to identifying promising targets and further help us understand the underlying biological framework and how it relates to survival outcomes, with the global objective of improving patient outcomes [@Karaesmen_2017].

