---
title: "Conclusion and Future Work"
author: "Abbas Rizvi"
date: "11/11/2018"
output: pdf_document  
editor_options: 
    chunk_output_type: console
---
    
```{r conc_setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


Modern science progresses through corroboration, computing, and collaboration. *Corroboration* by the need to continually seek answers and verify results. Reproducibility is necessary for robust ideas and expansion of knowledge. In genomics, traditional biomedical researchers often study one gene for one phenotype for numerous years, possibly decades. More often than not, these hypotheses propagate from a single study years ago or from using *a priori* information of a specific system in hypothesis development. Unfortunately, often, the original studies were never reproduced in well-designed independent studies. Constant pursuit of false leads may impede the rate at which important problems are solved. Agnostic approaches through usage of genome-wide technologies (such as GWAS), allow researchers to inspect phenotypes and gain biological insights at a global level. As technology continues to advance, data collection and data resources  inherently grow. Multiple rich data resources are available from all aspects of the genome, the *-omics* (i.e. genomics, transcriptomics, metabolomics, proteomics, methylation), as well as clinical trials. As such, large scale *computing* is imperative to the success of modern studies. Broader insights can be leveraged to develop hypotheses that can investigate and characterize complex disease etyiology. And the last element in order to accomplsih anything is *collaboration* by conducting team science. Bringing together multiple talented individuals and groups together and enriching each other with combined expertise to achieve a common goal. 

This dissertation was an attempt to exemplify these principles of modern science to the best of our ability. I was very fortunate to join a lab that had an incredibly rich clinically relevant data source, the DISCOVeRY-BMT GWAS. DISCOVeRY-BMT is a fine example of collaborative team science. The team is made up of scientists, clincians, and statisticians (and some who belong to more than one of those groups) from multiple institutions. With this GWAS, we attempted to corroborate researchers that continually pursued candidate SNPs in candidate genes that they believed held some biological relevance to survival in recipients or donors after transplant. We were unable to reproduce any of their results using DISCOVeRY BMT's much larger and more homogeneous study population. While we published on a negative result, we were able to make an important contribution to the transplant field by providing strong evidence that those studies should begin looking elsewhere in the genome for hidden insights. 

We were able to address a computational challenges in genomics, specifically when it comes to survival analysis. We developed an R/Bioconductor package, gwasurvivr, that can directly take data from online imputation web services. As large studies continue to get funded with well-documented follow up data -- survival studies will continue to become more prevalent. Gwasurvivr will be an important computational resource for the next several years. To date, over 200 unique IPs have downloaded gwasurvivr. We were also able to develop a pipeline that enhanced our workflow and reproducibility efforts. The pipeline should be expanded to automate annotation using SNiPA database and HaploReg. Construction of a centralized database (PostgreSQL database) is underway that will continue to enhance our workflow. 

We were able to use our own software and pipeline to perform many GWAS using DISCOVeRY-BMT. While we reported recipient and donor SNP associations to survival outcomes post-HSCT, there remains the entire DISCOVeRY-BMT 100 day censoring analyses for ALL (and our other disesase groups). These results have yet to be analyzed, curated, and reported. These results are very important and relevant to the transplant field as patients that die within the first 100 days may present very different genetic profiles than those who survive until at least 1 year. This research will continue after my departure from our lab and I believe it will be quite impactful.




