---
title: "nod2_analysis"
author: "Abbas Rizvi"
date: "October 24, 2016"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, eval=F)
```

# DISCOVeRY-BMT

NOD2 is a gene that has been frequently been studied in candidate gene studies looking at genetic variants from patients who have been treated with blood and marrow transplants (BMT). Oftentimes when NOD2 is studied, three SNPs frequently appear which have been labeled as SNP8 (rs2066844), SNP12 (rs2066845), and SNP13 (rs2066847). SNP8 and SNP12 were genotyped in in the DISCOVeRY-BMT GWAS and SNP13 was not genotyped in DISCOVeRY-BMT. In order to still consider this SNP, we searched for a SNP that may be in LD with rs2066847, and the best LD pair found in DISCOVeRY-BMT was rs146528649 ($r^{2}$ = 0.7422). All of the groups that studied NOD2, also chose the same design in the way they analyzed this SNP, such that only the patient-donor pairs that were homozygous wildtype for all 3 SNPs were deemed "wild type", and on the contrary, if any one variant was present in any of the 3 SNPs in either of the donor-recipient pair, they were deemed to have a variant. Survival analysis was conducted using this aforementioned classification method.  

Here we will discuss how we pre-processed the DISCOVeRY-BMT data to grab these SNPs of interest and how we implemented a similar composite scoring design and subsequent survival analysis.

In order to replicate these studies we will subset rs2066844, rs2066845 and rs146528649 from the imputed data. Each of these SNPs lie on `chr16` between the ranges of `50000000-55000000`. The imputed data was found on the UB supercomputer at the following location:

```{bash eval=F}
## File location:
/projects/rpci/lsuchest/lsuchest/Rserve/ImputeData/var/db/gwas/imputed_data/BMT093013_forImpute/BMT093013_forImpute.chr16-50000000-55000000.impute2 
```

We need to be able to extract this information into `R` so we can pull the SNPs of interest and their corresponding genotype probabilities for each sample. In order to this, we first need to convert the file into a VCF file. Also we need a file listing the SNPs in an unlabeled column vector (`nod2_snps.txt`). We will use qctools to convert `.impute2` to `.vcf`.   


```{bash eval=F}
module load qctool
qctool -g BMT093013_forImpute.chr16-50000000-55000000.impute2 -og /projects/rpci/lsuchest/abbasriz/candidate_gene/nod2_rep/nod2_rep_dosages.vcf -incl-rsids /projects/rpci/lsuchest/abbasriz/candidate_gene/nod2_rep/nod2_snps.txt
```

Now that we have the vcf file, we can use `VariantAnnotation` library to easily pull this data into `R`.

```{r}
library(VariantAnnotation)
## read in vcf file
vcf <- readVcf("/projects/rpci/lsuchest/abbasriz/candidate_gene/cg_haplotypes/nod2_rep/nod2_rep_dosages.vcf", genome="hg19")
vcf
```

Now we will grab the donor and recipients FID and IID. These correspond to the indices for the samples that we will subset.

```{r eval=F}
## grab unique identifiers of patients in different cohorts
library(data.table)
patients <- c("/projects/rpci/lsuchest/lsuchest/Rserve/BMT/genetic_data/R_c1_EA_FID_IID.txt",
              "/projects/rpci/lsuchest/lsuchest/Rserve/BMT/genetic_data/R_c2_EA_FID_IID.txt",
              "/projects/rpci/lsuchest/lsuchest/Rserve/BMT/genetic_data/D_c1_EA_FID_IID.txt",
              "/projects/rpci/lsuchest/lsuchest/Rserve/BMT/genetic_data/D_c2_EA_FID_IID.txt")

cohorts <- lapply(patients, fread)
files <- c()
for(i in 1:length(patients)) files[i] <- strsplit(patients, "/")[[i]][9]      
names(cohorts) <- files
head(cohorts)

rec.ids <- list(cohorts[[1]], cohorts[[2]])
names(rec.ids) <- files[1:2]
donor.ids <- list(cohorts[[3]], cohorts[[4]])
names(donor.ids) <- files[3:4]
## read in phenotype files files
r.pheno <- fread("/projects/rpci/lsuchest/lsuchest/Rserve/TheData/Plink_Recipient.pheno")
d.pheno <- fread("/projects/rpci/lsuchest/lsuchest/Rserve/TheData/Plink_Donor.pheno")
r.cov <- fread("/projects/rpci/lsuchest/lsuchest/Rserve/TheData/Plink_Recipient.cov")


## SUBSET EUROPEAN AMERICAN PATIENTS BASED OFF OF IID
id.subset <- function(x, pheno){pheno[pheno$IID %in% x$IID]}
# recipients
rec.pheno.ea <- lapply(rec.ids, id.subset, r.pheno)
# donor
don.pheno.ea <- lapply(donor.ids, id.subset, d.pheno)
# bring down to just FID, IID, pair_id
don.pheno.ea <- lapply(don.pheno.ea, function(x) x[,c("FID", "IID", "pair_id"),with=F])
don.pheno.ea <- lapply(don.pheno.ea, setnames, c("D_FID", "D_IID", "pair_id"))

# merge to recipient file

merge.pair_id <- function(recipient, donor){
        setkey(recipient, pair_id)
        setkey(donor, pair_id)
        recipient[donor]
}

merged.ea <- mapply(merge.pair_id, rec.pheno.ea, don.pheno.ea, SIMPLIFY=FALSE)

merged.ea <- data.table(do.call(rbind, merged.ea))

setnames(merged.ea, c("R_FID", "R_IID", colnames(merged.ea)[3:ncol(merged.ea)]))
# I think there are duplicated NAs because of the almost 2806 match and now 2815
merged.ea <- merged.ea[!duplicated(merged.ea$R_IID)]

ids <- merged.ea[,c("pair_id", "R_IID", "D_IID"), with=F]

# impute sample names

info.samples <- fread("/projects/rpci/lsuchest/lsuchest/Rserve/ImputeData/var/db/gwas/imputed_data/BMT093013_forImpute/BMT093013_forImpute.chr16-50000000-55000000.impute2_samples")

# top row is useless
info.samples <- info.samples[-1]

nrow(info.samples)
# 6805 samples
nrow(gt)
# nice! same number of samples...6805

# relabel the IDs so they just correspond to the sample order
# because when we converted the impute2 file to vcf, the order remains the same as seen info the impute2_samples file
# so we can just relabel the IDs as 1:6805 samples and go from there
info.samples$ID_1 <- seq(1, nrow(info.samples))


# okay now we will grab the indices from the impute_sample file and append them as columns to the ids data.table
ids$r_sample_index <- info.samples$ID_1[match(ids$R_IID, info.samples$ID_2)]
ids$d_sample_index <- info.samples$ID_1[match(ids$D_IID, info.samples$ID_2)]

# okay so now 22 donors didnt map back to that file
missing_samples <- ids[is.na(ids$d_sample_index)]
missing_samples
## weird .... but we will just omit them and move on because this is taking wayyyy too long and we'll ask lara about those samples later

ids <- na.omit(ids)
head(ids)
# now we have 2783 samples that we can go and grab all the genotype info on
```

Now we we will grab the genotype threshold values

```{r}
## threshold genotypes
# grab genotypes of 3 SNPs for all patients
gt <- geno(vcf)$GT
gt[1:nrow(gt),1:5]
```

We can see that we have both typed and imputed for rs2066844. We will keep only the imputed ones (which is the second rs2066844) 

```{r}
# only we have both typed and imputed of rs2066844
# so we remove the first rs2066844 because it is typed and we are keeping just the imputed ones
gt <- gt[-2,]
gt[1:nrow(gt),1:5]
```

Now we are going to transpose this data frame so we can have the samples as the rows and the SNPs as the columns

```{r}
# transpose the data.frame so we can patients as rows
gt <- data.frame(t(gt))
gt[1:5, 1:ncol(gt)]
```

Now we code the alleles in a dominant model. So if there is any `1` alleles, we will just relabel it as `1` and if there are homozygous `0`, we will relabel as `0`. We will also relabel the `.` as NA and then remove the NAs 

```{r eval=F}
## coding dominant model over df columns
gt <- apply(gt, 2, FUN=function(x) gsub("[.]", NA, x))
gt <- apply(gt, 2, FUN=function(x) gsub("0/0", 0, x))
gt <- apply(gt, 2, FUN=function(x) gsub("0/1", 1, x))
gt <- apply(gt, 2, FUN=function(x) gsub("1/1", 1, x))

## create a data.frame that changes the factors into numeric and keeps the row.names ...
gt <- data.frame(apply(gt, 2, function(x) as.numeric(as.character(x))), check.names=F, row.names=row.names(gt))
head(gt)
```

Now we will subset based off of the genome and cohort we are interested and create variables to do the composite SNP testing.

So if SNP 8 and SNP 12 had `0` for both SNPs, they will be recoded as `0`, and if any had `1` in it, it will be recoded as a `1`.
This will also done for the 3 SNPs (SNP8/SNP12/SNP13).

```{r eval=F}
## parse the unique IDs (genome/cohort) from the main genotype df into 4 dfs specific to genome/cohort
recs <- gt[ids$r_sample_index,]
donors <- gt[ids$d_sample_index,]

## recode with composite coding -- 0 if snps are all 0, 1 if any of the snps are 1
# snp 8 =  rs2066844 x[,2]
# snp 12 = rs2066845 x[,3]
# snp 13 = rs146528649 x[,1]
recode <- function(x){
        x$snp8snp12 <- ifelse((x[,2] == 0 & x[,3] == 0), x$snp8snp12 <- 0, x$snp8snp12 <- 1)
        x$compSNPs <- ifelse((x[,1] == 0 & x[,2] == 0 & x[,3] == 0), x$compSNPs <- 0, x$compSNPs <- 1)
        data.table(x, keep.rownames = T)
}
recs <- recode(recs)
donors <- recode(donors)

# remove "sample" from sample columns so we just have indices that we can map back to phenotype file
rm.sample <- function(x){ x$rn <- gsub("sample_", "", x$rn)
class(x$rn) <- "numeric"
x}

recs <- rm.sample(recs)
donors <- rm.sample(donors)

# match column names to ids column names
colnames(recs)[1] <- "r_sample_index"
colnames(donors)[1] <- "d_sample_index"

# grab pair ids so we can compare donor and recipient pairs
recs$pair_id <- ids$pair_id
donors$pair_id <- ids$pair_id

# function to recode 'score', again, if there is 0 in both D-R pairs the score 0, otherwise its 1
comp_score <- function(x){
	x$NOD2_score <- ifelse((x[,"recipient"] == 0 & x[,"donor"] == 0), x$NOD2_score <- 0, x$NOD2_score <- 1)
	x
}

# merge snp8/snp12 donor-recipient pair data.frame
snp8snp12.r <- recs[,c("snp8snp12", "pair_id")]
setkey(snp8snp12.r, pair_id)
snp8snp12.d <- donors[,c("snp8snp12", "pair_id")]
setkey(snp8snp12.d, pair_id)
snp8snp12 <- snp8snp12.d[snp8snp12.r]
colnames(snp8snp12) <- c("donor", "pair_id", "recipient")
snp8snp12 <- snp8snp12[,c("pair_id", "donor", "recipient")]
snp8snp12 <- comp_score(snp8snp12)

# merge snp8/snp12/snp13 donor-recipient pair data.frame
all3.r <- recs[,c("compSNPs", "pair_id")]
setkey(all3.r, pair_id)
all3.d <- donors[,c("compSNPs", "pair_id")]
setkey(all3.d, pair_id)
all3 <- all3.d[all3.r]
colnames(all3) <- c("donor", "pair_id", "recipient")
all3 <- all3[,c("pair_id", "donor", "recipient")]
all3 <- comp_score(all3)


colnames(snp8snp12)[ncol(snp8snp12)] <- "snp8snp12_score"
colnames(all3)[ncol(all3)] <- "all3_score"

snp8snp12 <- snp8snp12[,c("pair_id","snp8snp12_score")]
all3 <- all3[,c("pair_id", "all3_score")]
```

Now we have the recoded SNP8 and SNP12. We need to append these back to the phenotype files so we can run survival analyses on these. Here we grab the recipient and donor phenotype files and merge them based off of `pair_id` column. 
```{r eval=F}
# merge back to phenotype file .. which is called merged.ea
setkey(merged.ea, pair_id)
setkey(snp8snp12, pair_id)
setkey(all3, pair_id)

merged.ea <- merged.ea[snp8snp12]
merged.ea <- merged.ea[all3]

# lets quickly see how many people had NA genotype scores
table(is.na(merged.ea$all3_score))
# 486 deemed NA for all 3
table(is.na(merged.ea$snp8snp12_score))
# 208 deemed NA for snp8snp12
# write.table(merged.ea, file="pheno_merged_nod2haps.txt", sep="\t", row.names=F, col.names=T, quote=F)
```

# Survival Analysis
Okay now we can finally do survival....

```{r eval=F}

#######################
##### SURVIVAL ANALYSIS
#######################


## do survival
library(survival)

# grab covariate files
r.cov <- fread("/projects/rpci/lsuchest/lsuchest/Rserve/TheData/Plink_Recipient.cov")

r.cov <- r.cov[r.cov$pair_id %in% ids$pair_id]
setkey(r.cov, pair_id)
r.cov$cohort <- merged.ea$cohort

r.cov.c1 <- r.cov[cohort=="c1"]
r.cov.c2 <- r.cov[cohort=="c2"]


##################################
## DD covariates are: age, distatD
##################################


## cohort 1

# snp 8/snp12
DD.c1 <- Surv(time=merged.ea.c1$intxsurv_1Y, event=merged.ea.c1$disease_death_1Y==1)
dd.snp8snp12.c1 <- coxph(DD.c1~merged.ea.c1$snp8snp12_score+r.cov.c1$distatD)
dd.snp8snp12.c1.coef <- summary(dd.snp8snp12.c1)$coef[1,1]
dd.snp8snp12.c1.se.coef <- summary(dd.snp8snp12.c1)$coef[1,3]
dd.snp8snp12.c1.hr <- summary(dd.snp8snp12.c1)$coef[1,2]
dd.snp8snp12.c1.pval <- summary(dd.snp8snp12.c1)$coef[1,5] 

# snp8/snp12/snp13
dd.all3.c1 <- coxph(DD.c1~merged.ea.c1$all3_score+r.cov.c1$distatD)


## cohort 2
# snp 8/snp12
DD.c2 <- Surv(time=merged.ea.c2$intxsurv_1Y, event=merged.ea.c2$disease_death_1Y==1)
dd.snp8snp12.c2 <- coxph(DD.c2~merged.ea.c2$snp8snp12_score+r.cov.c2$distatD)


# snp8/snp12/snp13
dd.all3.c2 <- coxph(DD.c2~merged.ea.c2$all3_score+r.cov.c2$distatD)
dd.all3.c2 <- coxph(DD.c2~merged.ea.c2$all3_score+r.cov.c2$distatD)


###########################################
## OS covariates are: age, distatD, Pblood
###########################################
        
#### OS ####
# cohort 1
## OS covariates are: age, distatD, Pblood
OS.c1 <- Surv(time=merged.ea.c1$intxsurv_1Y, event=merged.ea.c1$dead_1Y==1)
# OS SNP8/SNP12
os.snp8snp12.c1 <- coxph(OS.c1~merged.ea.c1$snp8snp12_score+r.cov.c1$age+r.cov.c1$distatD+r.cov.c1$PBlood)
summary(os.snp8snp12.c1)$coef[1,5] 

# OS SNP8/SNP12/SNP13
os.all3.c1 <- coxph(OS.c1~merged.ea.c1$all3_score+r.cov.c1$age+r.cov.c1$distatD+r.cov.c1$PBlood)
summary(os.all3.c1)$coef[1,5] 


## cohort 2
OS.c2 <- Surv(time=merged.ea.c2$intxsurv_1Y, event=merged.ea.c2$dead_1Y==1)
# OS SNP8/SNP12
os.snp8snp12.c2 <- coxph(OS.c2~merged.ea.c2$snp8snp12_score+r.cov.c2$age+r.cov.c2$distatD+r.cov.c2$PBlood)
summary(os.snp8snp12.c2)$coef[1,5] 

# OS SNP8/SNP12/SNP13
os.all3.c2 <- coxph(OS.c2~merged.ea.c2$all3_score+r.cov.c2$age+r.cov.c2$distatD+r.cov.c2$PBlood)
summary(os.all3.c2)$coef[1,5] 

####################################
## PFS covariates: age, distatD
####################################
# cohort 1
PFS.c1 <- Surv(time=merged.ea.c1$intxsurv_1Y, event=merged.ea.c1$lfs_1Y==1)
# PFS SNP8 / SNP12
pfs.snp8snp12.c1 <- coxph(PFS.c1~merged.ea.c1$snp8snp12_score+r.cov.c1$distatD+r.cov.c1$age)
summary(pfs.snp8snp12.c1)$coefficients[1,5]
# PFS SNP 8 / SNP 12 / SNP 13
pfs.all3.c1 <- coxph(PFS.c1~as.numeric(merged.ea.c1$all3_score)+r.cov.c1$distatD+as.numeric(r.cov.c1$age))
summary(pfs.all3.c1)$coefficients[1,5]


# cohort 2
PFS.c2 <- Surv(time=merged.ea.c2$intxsurv_1Y, event=merged.ea.c2$lfs_1Y==1)
# PFS SNP8 / SNP12
pfs.snp8snp12.c2 <- coxph(PFS.c2~merged.ea.c2$snp8snp12_score+r.cov.c2$distatD+r.cov.c2$age)
summary(pfs.snp8snp12.c2)$coefficients[1,5]
# PFS SNP 8 / SNP 12 / SNP 13
pfs.all3.c2 <- coxph(PFS.c2~as.numeric(merged.ea.c2$all3_score)+r.cov.c2$distatD+as.numeric(r.cov.c2$age))
summary(pfs.all3.c2)$coefficients[1,5]





###############################################
# TRM covariates: age, bmiOBS, bmiOVWT, PBlood
##############################################

# cohort 1
TRM.c1 <- Surv(time=merged.ea.c1$intxsurv_1Y, event=merged.ea.c1$TRM_1Y==1)
# TRM SNP 8 / SNP 12
trm.snp8snp12.c1 <- coxph(TRM.c1~as.numeric(merged.ea.c1$snp8snp12_score)+as.numeric(r.cov.c1$age)+r.cov.c1$PBlood+r.cov.c1$bmiOBS+r.cov.c1$bmiOVWT)
summary(trm.snp8snp12.c1)$coefficients[1,5]
# TRM SNP 8 / SNP 12 / SNP 13
trm.all3.c1 <- coxph(TRM.c1~as.numeric(merged.ea.c1$all3_score)+as.numeric(r.cov.c1$age)+r.cov.c1$PBlood+as.numeric(r.cov.c1$bmiOBS)+as.numeric(r.cov.c1$bmiOVWT))
summary(trm.all3.c1)$coefficients[1,5]



# cohort 2
TRM.c2 <- Surv(time=merged.ea.c2$intxsurv_1Y, event=merged.ea.c2$TRM_1Y==1)
# TRM SNP 8 / SNP 12
trm.snp8snp12.c2 <- coxph(TRM.c2~as.numeric(merged.ea.c2$snp8snp12_score)+as.numeric(r.cov.c2$age)+r.cov.c2$PBlood+r.cov.c2$bmiOBS+r.cov.c2$bmiOVWT)
summary(trm.snp8snp12.c2)$coefficients[1,5]
# TRM SNP 8 / SNP 12 / SNP 13
trm.all3.c2 <- coxph(TRM.c2~as.numeric(merged.ea.c2$all3_score)+as.numeric(r.cov.c2$age)+r.cov.c2$PBlood+as.numeric(r.cov.c2$bmiOBS)+as.numeric(r.cov.c2$bmiOVWT))
summary(trm.all3.c2)$coefficients[1,5]


## WRITE TO TABLE
## NEED TO SPLIT INTO TWO TABLES FOR META-ANALYSIS
# will calculate confidence interval of hazard ratios AFTER
# set up columns so they are the same as our candidate gene table
cols <- c("gene",
          "rsID",
          "chr",
          "BP",
          "N",
          "allele1",
          "allele2",
          "coef",
          "se.coef",
          "exp.coef",
          "95%-CI",
          "Pvalue",
          "impute",
          "genome",
          "cohort",
          "outcome",
          "disease")


## Cohort 1

make.tbl <- function(outcome.vector, gene, rsID, chr, BP, N, allele1, allele2, genome, cohort, outcome, disease){
        coef <- outcome.vector[1,1]
        se.coef <- outcome.vector[1,3]
        hr <- outcome.vector[1,2]
        pval <- outcome.vector[1,5]
        res <- c(gene, rsID, chr, BP, N, allele1, allele2, coef, se.coef, hr, NA, pval, "imputed", genome, cohort, outcome, disease)
        res
}

# cohort 1
nod2.c1 <- data.table(rbind(make.tbl(summary(dd.snp8snp12.c1)$coef, "NOD2", "2 SNPs* DD", "chr16", "*", 2033, "*", "*", "S", "c1", "DD", "mixed"),
                            make.tbl(summary(os.snp8snp12.c1)$coef, "NOD2", "2 SNPs* OS", "chr16", "*", 2033, "*", "*", "S", "c1", "OS", "mixed"),
                            make.tbl(summary(pfs.snp8snp12.c1)$coef, "NOD2", "2 SNPs* PFS", "chr16", "*", 2033, "*", "*", "S", "c1", "PFS", "mixed"),
                            make.tbl(summary(trm.snp8snp12.c1)$coef, "NOD2", "2 SNPs* TRM", "chr16", "*", 2033, "*", "*", "S", "c1", "TRM", "mixed"),
                            make.tbl(summary(dd.all3.c1)$coef, "NOD2", "3 SNPs* DD", "chr16", "*", 2033, "*", "*", "S", "c1", "DD", "mixed"),
                            make.tbl(summary(os.all3.c1)$coef, "NOD2", "3 SNPs* OS", "chr16", "*", 2033, "*", "*", "S", "c1", "OS", "mixed"),
                            make.tbl(summary(pfs.all3.c1)$coef, "NOD2", "3 SNPs* PFS", "chr16", "*", 2033, "*", "*", "S", "c1", "PFS", "mixed"),
                            make.tbl(summary(trm.all3.c1)$coef, "NOD2", "3 SNPs* TRM", "chr16", "*", 2033, "*", "*", "S", "c1", "TRM", "mixed")))
colnames(nod2.c1) <- cols

write.table(nod2.c1, file="nod2.c1.txt", col.names=T, row.names=F, quote=F, sep='\t')
## Cohort 2
nod2.c2 <- data.table(rbind(make.tbl(summary(dd.snp8snp12.c2)$coef, "NOD2", "2 SNPs* DD", "chr16", "*", 757, "*", "*", "S", "c2", "DD", "mixed"),
                            make.tbl(summary(os.snp8snp12.c2)$coef, "NOD2", "2 SNPs* OS", "chr16", "*", 757, "*", "*", "S", "c2", "OS", "mixed"),
                            make.tbl(summary(pfs.snp8snp12.c2)$coef, "NOD2", "2 SNPs* PFS", "chr16", "*", 757, "*", "*", "S", "c2", "PFS", "mixed"),
                            make.tbl(summary(trm.snp8snp12.c2)$coef, "NOD2", "2 SNPs* TRM", "chr16", "*", 757, "*", "*", "S", "c2", "TRM", "mixed"),
                            make.tbl(summary(dd.all3.c2)$coef, "NOD2", "3 SNPs* DD", "chr16", "*", 757, "*", "*", "S", "c2", "DD", "mixed"),
                            make.tbl(summary(os.all3.c2)$coef, "NOD2", "3 SNPs* OS", "chr16", "*", 757, "*", "*", "S", "c2", "OS", "mixed"),
                            make.tbl(summary(pfs.all3.c2)$coef, "NOD2", "3 SNPs* PFS", "chr16", "*", 757, "*", "*", "S", "c2", "PFS", "mixed"),
                            make.tbl(summary(trm.all3.c2)$coef, "NOD2", "3 SNPs* TRM", "chr16", "*", 757, "*", "*", "S", "c2", "TRM", "mixed")))
colnames(nod2.c2) <- cols
write.table(nod2.c2, file="nod2.c2.txt", col.names=T, row.names=F, quote=F, sep='\t')


# RUN METAL 

## append meta results
nod2.meta <- fread("metal_S_M_mixed_nod21.tbl")
# remove direction column
nod2.meta <- nod2.meta[,-7]

# build up columns to match how our final table is set up
nod2.meta$gene <- "NOD2"
colnames(nod2.meta)[1] <- "rsID"
nod2.meta$chr <- "chr16"
nod2.meta$exp.coef <- exp(nod2.meta$Effect)
nod2.meta$genome <- "S"
nod2.meta$cohort <- "M"
nod2.meta$disease <- "mixed"
nod2.meta$N <- 2790
nod2.meta$BP <- "*"
nod2.meta$impute <- "imputed"
nod2.meta$`95%-CI` <- NA
nod2.meta$outcome <- NA
colnames(nod2.meta) <- c("rsID", "allele1", "allele2", "coef", "se.coef", "Pvalue", "gene", "chr", "exp.coef", "genome", "cohort", "disease", "N", "BP", "impute", "95%-CI", "outcome")

nod2.meta$outcome[c(1,8)] <- "OS"
nod2.meta$outcome[2:3] <- "DD"
nod2.meta$outcome[c(4,6)] <- "PFS"
nod2.meta$outcome[c(5,7)] <- "TRM"

setcolorder(nod2.meta, cols)

# join all of the nod2 results (c1, c2, meta analyiss)
nod2.final <- do.call(rbind, list(nod2.c1, nod2.c2, nod2.meta))

# remove extra names in rsID column

nod2.final$rsID <- substr(nod2.final$rsID, 1, 7)

## now calculate 95% confidence interval for hazard ratio
hr.ci <- function(coef.est, se){
        lb <- round(exp(coef.est-1.96*se), 5)
        ub <- round(exp(coef.est+1.96*se), 5)
        paste0("[", lb,", " , ub ,"]")
}

for(i in 1:nrow(nod2.final)){
        nod2.final[i,"95%-CI"] <- hr.ci(as.numeric(nod2.final$coef)[i], as.numeric(nod2.final$se.coef)[i])
}

write.table(nod2.final, file="nod2_survival_pvals.txt", col.names=T, row.names=F, sep="\t", quote=F)
```








